{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "55a6d898-df28-42bb-9852-b219ee364309",
   "metadata": {},
   "source": [
    "This demonstrates that pyld does not fully implement '@always' embedding (or at least, that I cannot figure out how to make it work...)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "312f8615-8057-4218-922e-71644e93e39f",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pyld import jsonld\n",
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "79e4e86f-d8a5-4785-a9cb-d25f84e3d88e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "input = json.loads('''{\n",
    "  \"id\": \"cggv:d9b3a56b-b188-44d1-9983-a059895c0ba1v4.0\",\n",
    "  \"type\": \"EvidenceStrengthAssertion\",\n",
    "  \"calculatedEvidenceStrength\": \"Definitive\",\n",
    "  \"evidence\": [\n",
    "    {\n",
    "      \"id\": \"cggv:d9b3a56b-b188-44d1-9983-a059895c0ba1_overall_genetic_evidence_line\",\n",
    "      \"type\": \"EvidenceLine\",\n",
    "      \"evidence\": [\n",
    "        {\n",
    "          \"id\": \"cggv:98e0665d-940d-4efb-b547-940d52a22e24_proband_score_evidence_line\",\n",
    "          \"type\": \"EvidenceLine\",\n",
    "          \"direction\": \"Inconclusive\",\n",
    "          \"evidence\": [\n",
    "            {\n",
    "              \"id\": \"cggv:98e0665d-940d-4efb-b547-940d52a22e24\",\n",
    "              \"type\": \"Proband\",\n",
    "              \"ageType\": \"AgeAtDeath\",\n",
    "              \"ageUnit\": \"Months\",\n",
    "              \"ageValue\": 11,\n",
    "              \"allele\": [\n",
    "                {\n",
    "                  \"id\": \"cggv:80517393-27f0-42a7-a1e7-2ee4596f1fcb\",\n",
    "                  \"type\": \"https://terms.ga4gh.org/VariationDescriptor\",\n",
    "                  \"http://www.w3.org/2004/02/skos/core#prefLabel\": \"NC_000020.11:g.34295508G>A (GRCh38)\",\n",
    "                  \"https://terms.ga4gh.org/CanonicalReference\": {\n",
    "                    \"id\": \"http://reg.genome.network/allele/CA408660970\"\n",
    "                  }\n",
    "                },\n",
    "                {\n",
    "                  \"id\": \"cggv:bf1cc479-73a2-4502-b116-5978f58028a8\",\n",
    "                  \"type\": \"https://terms.ga4gh.org/VariationDescriptor\",\n",
    "                  \"http://www.w3.org/2004/02/skos/core#prefLabel\": \"NC_000020.11:g.34295472C>T (GRCh38)\",\n",
    "                  \"https://terms.ga4gh.org/CanonicalReference\": {\n",
    "                    \"id\": \"http://reg.genome.network/allele/CA408660902\"\n",
    "                  }\n",
    "                }\n",
    "              ],\n",
    "              \"firstTestingMethod\": \"Whole genome shotgun sequencing\",\n",
    "              \"phenotypeFreeText\": \"plasma levels of methionine at 292 μmol/L (9–42 μmol/L), SAM at 5030 nM (33–95 nM), and SAH at 3290 nM (13–28 nM), CK of 1555 U/L (<355)\",\n",
    "              \"phenotypes\": [\n",
    "                \"obo:HP_0002878\",\n",
    "                \"obo:HP_0012498\",\n",
    "                \"obo:HP_0001612\",\n",
    "                \"obo:HP_0001252\",\n",
    "                \"obo:HP_0002910\",\n",
    "                \"obo:HP_0003236\",\n",
    "                \"obo:HP_0001410\",\n",
    "                \"obo:HP_0003235\",\n",
    "                \"obo:HP_0003073\",\n",
    "                \"obo:HP_0001561\",\n",
    "                \"obo:HP_0030714\",\n",
    "                \"obo:HP_0002104\"\n",
    "              ],\n",
    "              \"sex\": \"Male\",\n",
    "              \"variant\": [\n",
    "                {\n",
    "                  \"id\": \"cggv:39ec98c6-2ef0-4972-a043-02617a473186_variant_evidence_item\",\n",
    "                  \"type\": \"obo:SEPIO_0004116\",\n",
    "                  \"allele\": {\n",
    "                    \"id\": \"cggv:bf1cc479-73a2-4502-b116-5978f58028a8\"\n",
    "                  },\n",
    "                  \"alleleOrigin\": {\n",
    "                    \"id\": \"obo:GENO_0000877\"\n",
    "                  },\n",
    "                  \"zygosity\": {\n",
    "                    \"id\": \"cg:TwoVariantsInTrans\"\n",
    "                  },\n",
    "                  \"dc:source\": {\n",
    "                    \"id\": \"https://pubmed.ncbi.nlm.nih.gov/35789945\",\n",
    "                    \"type\": \"dc:BibliographicResource\",\n",
    "                    \"dc:abstract\": \"S-Adenosylhomocysteine (SAH) hydrolase deficiency is an autosomal recessive disorder in methionine metabolism caused by pathogenic variants in the gene \",\n",
    "                    \"dc:creator\": \"Huang Y\",\n",
    "                    \"dc:date\": \"2022\",\n",
    "                    \"dc:title\": \"The biochemical profile and dietary management in S-adenosylhomocysteine hydrolase deficiency.\"\n",
    "                  }\n",
    "                },\n",
    "                {\n",
    "                  \"id\": \"cggv:f56bd253-0e7c-4b28-af73-1b677a121a64_variant_evidence_item\",\n",
    "                  \"type\": \"obo:SEPIO_0004116\",\n",
    "                  \"allele\": {\n",
    "                    \"id\": \"cggv:80517393-27f0-42a7-a1e7-2ee4596f1fcb\"\n",
    "                  },\n",
    "                  \"alleleOrigin\": {\n",
    "                    \"id\": \"obo:GENO_0000877\"\n",
    "                  },\n",
    "                  \"zygosity\": {\n",
    "                    \"id\": \"cg:TwoVariantsInTrans\"\n",
    "                  },\n",
    "                  \"dc:source\": \"https://pubmed.ncbi.nlm.nih.gov/35789945\"\n",
    "                }\n",
    "              ],\n",
    "              \"rdfs:label\": \"Huang Proband\"\n",
    "            },\n",
    "            {\n",
    "              \"id\": \"cggv:39ec98c6-2ef0-4972-a043-02617a473186\",\n",
    "              \"type\": \"obo:SEPIO_0004097\",\n",
    "              \"calculatedScore\": 0.1,\n",
    "              \"direction\": \"Inconclusive\",\n",
    "              \"evidence\": [\n",
    "                {\n",
    "                  \"id\": \"cggv:39ec98c6-2ef0-4972-a043-02617a473186_variant_evidence_item\"\n",
    "                }\n",
    "              ],\n",
    "              \"strengthScore\": 0\n",
    "            },\n",
    "            {\n",
    "              \"id\": \"cggv:f56bd253-0e7c-4b28-af73-1b677a121a64\",\n",
    "              \"type\": \"obo:SEPIO_0004097\",\n",
    "              \"calculatedScore\": 0.1,\n",
    "              \"direction\": \"Inconclusive\",\n",
    "              \"evidence\": [\n",
    "                {\n",
    "                  \"id\": \"cggv:f56bd253-0e7c-4b28-af73-1b677a121a64_variant_evidence_item\"\n",
    "                }\n",
    "              ],\n",
    "              \"strengthScore\": 0\n",
    "            }\n",
    "          ],\n",
    "          \"specifiedBy\": \"GeneValidityMaximumProbandScoreCriteria\"\n",
    "        },\n",
    "        {\n",
    "          \"id\": \"cggv:788c32e5-57a0-48b9-8549-b2ecdbacef22_proband_score_evidence_line\",\n",
    "          \"type\": \"EvidenceLine\",\n",
    "          \"direction\": \"Supports\",\n",
    "          \"evidence\": [\n",
    "            {\n",
    "              \"id\": \"cggv:788c32e5-57a0-48b9-8549-b2ecdbacef22\",\n",
    "              \"type\": \"Proband\",\n",
    "              \"ageType\": \"AgeAtReport\",\n",
    "              \"ageUnit\": \"Months\",\n",
    "              \"ageValue\": 2,\n",
    "              \"variant\": [\n",
    "                {\n",
    "                  \"id\": \"cggv:8c54f7dd-e801-497f-ae03-11b7d59b3477_variant_evidence_item\",\n",
    "                  \"type\": \"obo:SEPIO_0004116\",\n",
    "                  \"allele\": {\n",
    "                    \"id\": \"cggv:f2cb1aab-a178-4938-9443-a47386171005\"\n",
    "                  },\n",
    "                  \"alleleOrigin\": {\n",
    "                    \"id\": \"obo:GENO_0000888\"\n",
    "                  },\n",
    "                  \"zygosity\": {\n",
    "                    \"id\": \"cg:TwoVariantsInTrans\"\n",
    "                  },\n",
    "                  \"dc:source\": {\n",
    "                    \"id\": \"https://pubmed.ncbi.nlm.nih.gov/27848944\",\n",
    "                    \"type\": \"dc:BibliographicResource\",\n",
    "                    \"dc:abstract\": \"We report our results of 1000 diagnostic WES cases based on 2819 sequenced samples from 54 countries with a wide phenotypic spectrum. Clinical information given by the requesting physicians was translated to HPO terms. WES processes were performed according to standardized settings. We identified the underlying pathogenic or likely pathogenic variants in 307 families (30.7%). In further 253 families (25.3%) a variant of unknown significance, possibly explaining the clinical symptoms of the index patient was identified. WES enabled timely diagnosing of genetic diseases, validation of causality of specific genetic disorders of PTPN23, KCTD3, SCN3A, PPOX, FRMPD4, and SCN1B, and setting dual diagnoses by detecting two causative variants in distinct genes in the same patient. We observed a better diagnostic yield in consanguineous families, in severe and in syndromic phenotypes. Our results suggest that WES has a better yield in patients that present with several symptoms, rather than an isolated abnormality. We also validate the clinical benefit of WES as an effective diagnostic tool, particularly in nonspecific or heterogeneous phenotypes. We recommend WES as a first-line diagnostic in all cases without a clear differential diagnosis, to facilitate personal medical care.\",\n",
    "                    \"dc:creator\": \"Trujillano D\",\n",
    "                    \"dc:date\": \"2017\",\n",
    "                    \"dc:title\": \"Clinical exome sequencing: results from 2819 samples reflecting 1000 families.\"\n",
    "                  }\n",
    "                },\n",
    "                {\n",
    "                  \"id\": \"cggv:c0a9d02b-d48a-4ab2-8859-22ddb557e4b0_variant_evidence_item\",\n",
    "                  \"type\": \"obo:SEPIO_0004116\",\n",
    "                  \"allele\": {\n",
    "                    \"id\": \"cggv:79b7760a-855c-4ff4-a22e-49d891888cb3\"\n",
    "                  },\n",
    "                  \"alleleOrigin\": {\n",
    "                    \"id\": \"obo:GENO_0000888\"\n",
    "                  },\n",
    "                  \"zygosity\": {\n",
    "                    \"id\": \"cg:TwoVariantsInTrans\"\n",
    "                  },\n",
    "                  \"dc:source\": \"https://pubmed.ncbi.nlm.nih.gov/27848944\"\n",
    "                }\n",
    "              ],\n",
    "              \"rdfs:label\": \"00080802\"\n",
    "            },\n",
    "            {\n",
    "              \"id\": \"cggv:8c54f7dd-e801-497f-ae03-11b7d59b3477\",\n",
    "              \"type\": \"obo:SEPIO_0004097\",\n",
    "              \"calculatedScore\": 0.5,\n",
    "              \"direction\": \"Supports\",\n",
    "              \"evidence\": [\n",
    "                {\n",
    "                  \"id\": \"cggv:8c54f7dd-e801-497f-ae03-11b7d59b3477_variant_evidence_item\"\n",
    "                },\n",
    "                {\n",
    "                  \"id\": \"cggv:8c54f7dd-e801-497f-ae03-11b7d59b3477_function_evidence_item\",\n",
    "                  \"type\": \"VariantFunctionalImpactEvidence\",\n",
    "                  \"functionalDataSupport\": \"Yes\",\n",
    "                  \"dc:description\": \"The maternally inherited Tyr143Cys missense variant has been observed several times since the initial report in PMID: 15024124 and was functionally characterized in PMID: 16872278.\"\n",
    "                }\n",
    "              ],\n",
    "              \"strengthScore\": 0.25,\n",
    "              \"dc:description\": \"Methionine levels were not reported nor was enzymatic activity in patient cells.\"\n",
    "            }\n",
    "          ],\n",
    "          \"specifiedBy\": \"GeneValidityMaximumProbandScoreCriteria\"\n",
    "        },\n",
    "        {\n",
    "          \"id\": \"cggv:cf77708f-926f-4275-84fa-49fc7b4f894e_proband_score_evidence_line\",\n",
    "          \"type\": \"EvidenceLine\",\n",
    "          \"direction\": \"Supports\",\n",
    "          \"evidence\": [\n",
    "            {\n",
    "              \"id\": \"cggv:cf77708f-926f-4275-84fa-49fc7b4f894e\",\n",
    "              \"type\": \"Proband\",\n",
    "              \"allele\": [\n",
    "                {\n",
    "                  \"id\": \"cggv:79b7760a-855c-4ff4-a22e-49d891888cb3\"\n",
    "                },\n",
    "                {\n",
    "                  \"id\": \"cggv:f2cb1aab-a178-4938-9443-a47386171005\"\n",
    "                }\n",
    "              ],\n",
    "              \"sex\": \"Male\",\n",
    "              \"variant\": [\n",
    "                {\n",
    "                  \"id\": \"cggv:d84cd273-0720-44c6-9cec-67dd878b434c_variant_evidence_item\",\n",
    "                  \"type\": \"obo:SEPIO_0004116\",\n",
    "                  \"allele\": {\n",
    "                    \"id\": \"cggv:79b7760a-855c-4ff4-a22e-49d891888cb3\"\n",
    "                  },\n",
    "                  \"alleleOrigin\": {\n",
    "                    \"id\": \"obo:GENO_0000880\"\n",
    "                  },\n",
    "                  \"paternityMaternityConfirmed\": \"No\",\n",
    "                  \"zygosity\": {\n",
    "                    \"id\": \"cg:TwoVariantsInTrans\"\n",
    "                  },\n",
    "                  \"dc:source\": {\n",
    "                    \"id\": \"https://pubmed.ncbi.nlm.nih.gov/16736098\",\n",
    "                    \"type\": \"dc:BibliographicResource\",\n",
    "                    \"dc:abstract\": \"This paper reports the third proven human case of deficient S-adenosylhomocysteine (AdoHcy) hydrolase activity. The patient is similar to the only two previously reported cases with this disorder in having severe myopathy, developmental delay, elevated serum creatine kinase (CK) concentrations, and hypermethioninaemia. Although he has been followed from infancy, the basic enzyme deficiency was established only at age 26 years. The diagnosis was based on markedly elevated plasma concentrations of both AdoHcy and S-adenosylmethionine, some 20% of the mean control activity of AdoHcy hydrolase activity in haemolysates of his red-blood cells, and two missense mutations in his gene encoding AdoHcy hydrolase. He had low values of erythrocyte phosphatidylcholine and plasma free choline and marginally elevated excretion of guanidinoacetate, suggesting that the elevated AdoHcy may have been inhibiting methylation of phosphatidylethanolamine and guanidinoacetate. His leukocyte DNA was globally more methylated than the DNA's of his parents or the mean extent of methylation measured in age-matched control subjects.\",\n",
    "                    \"dc:creator\": \"Buist NR\",\n",
    "                    \"dc:date\": \"2006\",\n",
    "                    \"dc:title\": \"S-adenosylhomocysteine hydrolase deficiency in a 26-year-old man.\"\n",
    "                  }\n",
    "                },\n",
    "                {\n",
    "                  \"id\": \"cggv:0fb58cb2-f500-484b-a91e-15cb473995d5_variant_evidence_item\",\n",
    "                  \"type\": \"obo:SEPIO_0004116\",\n",
    "                  \"allele\": {\n",
    "                    \"id\": \"cggv:f2cb1aab-a178-4938-9443-a47386171005\"\n",
    "                  },\n",
    "                  \"alleleOrigin\": {\n",
    "                    \"id\": \"obo:GENO_0000888\"\n",
    "                  },\n",
    "                  \"zygosity\": {\n",
    "                    \"id\": \"cg:TwoVariantsInTrans\"\n",
    "                  },\n",
    "                  \"dc:source\": \"https://pubmed.ncbi.nlm.nih.gov/16736098\"\n",
    "                }\n",
    "              ],\n",
    "              \"rdfs:label\": \"Buist Case\"\n",
    "            },\n",
    "            {\n",
    "              \"id\": \"cggv:d84cd273-0720-44c6-9cec-67dd878b434c\",\n",
    "              \"type\": \"obo:SEPIO_0004097\",\n",
    "              \"calculatedScore\": 1,\n",
    "              \"direction\": \"Supports\",\n",
    "              \"evidence\": [\n",
    "                {\n",
    "                  \"id\": \"cggv:d84cd273-0720-44c6-9cec-67dd878b434c_variant_evidence_item\"\n",
    "                },\n",
    "                {\n",
    "                  \"id\": \"cggv:d84cd273-0720-44c6-9cec-67dd878b434c_function_evidence_item\",\n",
    "                  \"type\": \"VariantFunctionalImpactEvidence\",\n",
    "                  \"functionalDataSupport\": \"Yes\",\n",
    "                  \"dc:description\": \"Functionally characterized in PMID: 18211827; in e. coli, expression of recombinant protein showed that the catalytic rates of Ala89Val are decreased at least 70% compared to WT and it was thermosensitive, with an unfolding temperature significantly reduced. \"\n",
    "                }\n",
    "              ],\n",
    "              \"strengthScore\": 1,\n",
    "              \"dc:description\": \"Ala89Val occurs at a 0.00003531 MAF in the gnomAD non-Finnish European population.\"\n",
    "            }\n",
    "          ],\n",
    "          \"specifiedBy\": \"GeneValidityMaximumProbandScoreCriteria\"\n",
    "        },\n",
    "        {\n",
    "          \"id\": \"cggv:3566ab7e-0541-4def-9f45-4f9dc45fa0dd_proband_score_evidence_line\",\n",
    "          \"type\": \"EvidenceLine\",\n",
    "          \"direction\": \"Supports\",\n",
    "          \"evidence\": [\n",
    "            {\n",
    "              \"id\": \"cggv:3566ab7e-0541-4def-9f45-4f9dc45fa0dd\",\n",
    "              \"type\": \"Proband\",\n",
    "              \"ageType\": \"AgeAtDeath\",\n",
    "              \"ageUnit\": \"Days\",\n",
    "              \"ageValue\": 95,\n",
    "              \"allele\": [\n",
    "                {\n",
    "                  \"id\": \"cggv:da6d5475-0147-4dc7-88a7-b14014c51784\",\n",
    "                  \"type\": \"https://terms.ga4gh.org/VariationDescriptor\",\n",
    "                  \"http://www.w3.org/2004/02/skos/core#prefLabel\": \"NC_000020.11:g.34295444G>A (GRCh38)\",\n",
    "                  \"https://terms.ga4gh.org/CanonicalReference\": {\n",
    "                    \"id\": \"http://reg.genome.network/allele/CA408660843\"\n",
    "                  }\n",
    "                },\n",
    "                {\n",
    "                  \"id\": \"cggv:fb426440-50ba-4162-b340-0d512f9dfd0a\",\n",
    "                  \"type\": \"https://terms.ga4gh.org/VariationDescriptor\",\n",
    "                  \"http://www.w3.org/2004/02/skos/core#prefLabel\": \"NC_000020.11:g.34290848C>T (GRCh38)\",\n",
    "                  \"https://terms.ga4gh.org/CanonicalReference\": {\n",
    "                    \"id\": \"http://reg.genome.network/allele/CA408657898\"\n",
    "                  }\n",
    "                }\n",
    "              ],\n",
    "              \"previousTestingDescription\": \"Organic acid analysis gave normal results and the serum methionine level (19.89 mmol/L) was in normal range on DOL 41 (reference: 6–58.80 mmol/L). \",\n",
    "              \"secondTestingMethod\": \"Sanger sequencing\",\n",
    "              \"sex\": \"Female\",\n",
    "              \"variant\": [\n",
    "                {\n",
    "                  \"id\": \"cggv:a72c84f0-0878-4134-b7c1-3bbe85061b30_variant_evidence_item\",\n",
    "                  \"type\": \"obo:SEPIO_0004116\",\n",
    "                  \"allele\": {\n",
    "                    \"id\": \"cggv:fb426440-50ba-4162-b340-0d512f9dfd0a\"\n",
    "                  },\n",
    "                  \"alleleOrigin\": {\n",
    "                    \"id\": \"obo:GENO_0000888\"\n",
    "                  },\n",
    "                  \"zygosity\": {\n",
    "                    \"id\": \"cg:TwoVariantsInTrans\"\n",
    "                  },\n",
    "                  \"dc:source\": {\n",
    "                    \"id\": \"https://pubmed.ncbi.nlm.nih.gov/31957987\",\n",
    "                    \"type\": \"dc:BibliographicResource\",\n",
    "                    \"dc:abstract\": \"S-adenosylhomocysteine hydrolase deficiency is an autosomal recessive neurometabolic disorder affecting the muscles, liver, and nervous system. The disease occurs by pathogenic variants of AHCY gene encoding S-adenosylhomocysteine hydrolase (AHCY) enzyme. This article reports a patient with presumed AHCY deficiency who was diagnosed by whole exome sequencing due to compound heterozygosity of novel p.T57I (c.170C>T) and p.V217M (c.649G>A) variants of AHCY gene. The patient had diffuse edema, coagulopathy, central nervous system abnormalities, and hypotonia. She died in 3 months due to cardiovascular collapse. Clinical findings of the present case were compatible with previously reported AHCY deficiency patients and the novel variants we found are considered to be the cause of the symptoms. This article also compiles the previous reports and expands clinical spectrum of AHCY deficiency by adding new features.\",\n",
    "                    \"dc:creator\": \"Bas H\",\n",
    "                    \"dc:date\": \"2020\",\n",
    "                    \"dc:title\": \"A Turkish patient with novel AHCY variants and presumed diagnosis of S-adenosylhomocysteine hydrolase deficiency.\"\n",
    "                  }\n",
    "                },\n",
    "                {\n",
    "                  \"id\": \"cggv:109ef32f-7d2b-4ee3-b4b0-1de826fbfc12_variant_evidence_item\",\n",
    "                  \"type\": \"obo:SEPIO_0004116\",\n",
    "                  \"allele\": {\n",
    "                    \"id\": \"cggv:da6d5475-0147-4dc7-88a7-b14014c51784\"\n",
    "                  },\n",
    "                  \"alleleOrigin\": {\n",
    "                    \"id\": \"obo:GENO_0000888\"\n",
    "                  },\n",
    "                  \"zygosity\": {\n",
    "                    \"id\": \"cg:TwoVariantsInTrans\"\n",
    "                  },\n",
    "                  \"dc:source\": \"https://pubmed.ncbi.nlm.nih.gov/31957987\"\n",
    "                }\n",
    "              ],\n",
    "              \"rdfs:label\": \"Bas patient\"\n",
    "            }\n",
    "          ],\n",
    "          \"specifiedBy\": \"GeneValidityMaximumProbandScoreCriteria\"\n",
    "        }\n",
    "      ],\n",
    "      \"specifiedBy\": \"GeneValidityOverallGeneticEvidenceCriteria\",\n",
    "      \"strengthScore\": 8.95\n",
    "    }\n",
    "  ],\n",
    "  \"evidenceStrength\": \"Definitive\",\n",
    "  \"sequence\": 9529,\n",
    "  \"specifiedBy\": \"GeneValidityCriteria11\",\n",
    "  \"strengthScore\": 11.95,\n",
    "  \"subject\": {\n",
    "    \"id\": \"cggv:2b13396b-44f2-427b-b0c6-849203bb2d89\",\n",
    "    \"type\": \"GeneValidityProposition\",\n",
    "    \"disease\": \"obo:MONDO_0013404\",\n",
    "    \"gene\": \"hgnc:343\",\n",
    "    \"modeOfInheritance\": \"obo:HP_0000007\"\n",
    "  },\n",
    "  \"version\": \"4.0\",\n",
    "  \"dc:isVersionOf\": {\n",
    "    \"id\": \"cggv:d9b3a56b-b188-44d1-9983-a059895c0ba1\"\n",
    "  },\n",
    "  \"@context\": {\n",
    "    \"@vocab\": \"http://dataexchange.clinicalgenome.org/terms/\",\n",
    "    \"id\": \"@id\",\n",
    "    \"type\": \"@type\",\n",
    "    \"cg\": \"http://dataexchange.clinicalgenome.org/terms/\",\n",
    "    \"dc\": \"http://purl.org/dc/terms/\",\n",
    "    \"obo\": \"http://purl.obolibrary.org/obo/\",\n",
    "    \"hgnc\": \"https://identifiers.org/hgnc:\",\n",
    "    \"rdfs\": \"http://www.w3.org/2000/01/rdf-schema#\",\n",
    "    \"cggv\": \"http://dataexchange.clinicalgenome.org/gci/\",\n",
    "    \"evidenceStrength\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"calculatedEvidenceStrength\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"computedEvidenceStrength\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"specifiedBy\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"evidence\": {\n",
    "      \"@container\": \"@set\"\n",
    "    },\n",
    "    \"disease\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"gene\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"phenotypes\": {\n",
    "      \"@type\": \"@id\"\n",
    "    },\n",
    "    \"agent\": {\n",
    "      \"@type\": \"@id\"\n",
    "    },\n",
    "    \"role\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"modeOfInheritance\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"sex\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"direction\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"ageType\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"ageUnit\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"dc:source\": {\n",
    "      \"@type\": \"@id\"\n",
    "    }\n",
    "  }\n",
    "}''')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "31c6fc30-aa2d-4b7a-81bf-bc1130e95dc7",
   "metadata": {},
   "outputs": [],
   "source": [
    "frame = json.loads('''{\n",
    "  \"@context\": {\n",
    "    \"@vocab\": \"http://dataexchange.clinicalgenome.org/terms/\",\n",
    "    \"id\": \"@id\",\n",
    "    \"type\": \"@type\",\n",
    "    \"cg\": \"http://dataexchange.clinicalgenome.org/terms/\",\n",
    "    \"dc\": \"http://purl.org/dc/terms/\",\n",
    "    \"obo\": \"http://purl.obolibrary.org/obo/\",\n",
    "    \"hgnc\": \"https://identifiers.org/hgnc:\",\n",
    "    \"rdfs\": \"http://www.w3.org/2000/01/rdf-schema#\",\n",
    "    \"cggv\": \"http://dataexchange.clinicalgenome.org/gci/\",\n",
    "    \"evidenceStrength\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"calculatedEvidenceStrength\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"computedEvidenceStrength\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"specifiedBy\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"evidence\": {\n",
    "      \"@container\": \"@set\"\n",
    "    },\n",
    "    \"disease\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"gene\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"phenotypes\": {\n",
    "      \"@type\": \"@id\"\n",
    "    },\n",
    "    \"agent\": {\n",
    "      \"@type\": \"@id\"\n",
    "    },\n",
    "    \"role\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"modeOfInheritance\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"sex\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"direction\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"ageType\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"ageUnit\": {\n",
    "      \"@type\": \"@vocab\"\n",
    "    },\n",
    "    \"dc:source\": {\n",
    "      \"@type\": \"@id\"\n",
    "    }\n",
    "  },\n",
    "  \"@type\": \"Proband\",\n",
    "  \"@embed\": \"@always\"\n",
    "}''')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "8f33bb7c-a612-4008-84f0-89e50ad9cac2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'@context': {'@vocab': 'http://dataexchange.clinicalgenome.org/terms/',\n",
       "  'id': '@id',\n",
       "  'type': '@type',\n",
       "  'cg': 'http://dataexchange.clinicalgenome.org/terms/',\n",
       "  'dc': 'http://purl.org/dc/terms/',\n",
       "  'obo': 'http://purl.obolibrary.org/obo/',\n",
       "  'hgnc': 'https://identifiers.org/hgnc:',\n",
       "  'rdfs': 'http://www.w3.org/2000/01/rdf-schema#',\n",
       "  'cggv': 'http://dataexchange.clinicalgenome.org/gci/',\n",
       "  'evidenceStrength': {'@type': '@vocab'},\n",
       "  'calculatedEvidenceStrength': {'@type': '@vocab'},\n",
       "  'computedEvidenceStrength': {'@type': '@vocab'},\n",
       "  'specifiedBy': {'@type': '@vocab'},\n",
       "  'evidence': {'@container': '@set'},\n",
       "  'disease': {'@type': '@vocab'},\n",
       "  'gene': {'@type': '@vocab'},\n",
       "  'phenotypes': {'@type': '@id'},\n",
       "  'agent': {'@type': '@id'},\n",
       "  'role': {'@type': '@vocab'},\n",
       "  'modeOfInheritance': {'@type': '@vocab'},\n",
       "  'sex': {'@type': '@vocab'},\n",
       "  'direction': {'@type': '@vocab'},\n",
       "  'ageType': {'@type': '@vocab'},\n",
       "  'ageUnit': {'@type': '@vocab'},\n",
       "  'dc:source': {'@type': '@id'}},\n",
       " '@graph': [{'id': 'cggv:3566ab7e-0541-4def-9f45-4f9dc45fa0dd',\n",
       "   'type': 'Proband',\n",
       "   'ageType': 'AgeAtDeath',\n",
       "   'ageUnit': 'Days',\n",
       "   'ageValue': 95,\n",
       "   'allele': [{'id': 'cggv:da6d5475-0147-4dc7-88a7-b14014c51784',\n",
       "     'type': 'https://terms.ga4gh.org/VariationDescriptor',\n",
       "     'http://www.w3.org/2004/02/skos/core#prefLabel': 'NC_000020.11:g.34295444G>A (GRCh38)',\n",
       "     'https://terms.ga4gh.org/CanonicalReference': {'id': 'http://reg.genome.network/allele/CA408660843'}},\n",
       "    {'id': 'cggv:fb426440-50ba-4162-b340-0d512f9dfd0a',\n",
       "     'type': 'https://terms.ga4gh.org/VariationDescriptor',\n",
       "     'http://www.w3.org/2004/02/skos/core#prefLabel': 'NC_000020.11:g.34290848C>T (GRCh38)',\n",
       "     'https://terms.ga4gh.org/CanonicalReference': {'id': 'http://reg.genome.network/allele/CA408657898'}}],\n",
       "   'previousTestingDescription': 'Organic acid analysis gave normal results and the serum methionine level (19.89\\u2009mmol/L) was in normal range on DOL 41 (reference: 6–58.80\\u2009mmol/L). ',\n",
       "   'secondTestingMethod': 'Sanger sequencing',\n",
       "   'sex': 'Female',\n",
       "   'variant': [{'id': 'cggv:a72c84f0-0878-4134-b7c1-3bbe85061b30_variant_evidence_item',\n",
       "     'type': 'obo:SEPIO_0004116',\n",
       "     'allele': {'id': 'cggv:fb426440-50ba-4162-b340-0d512f9dfd0a',\n",
       "      'type': 'https://terms.ga4gh.org/VariationDescriptor',\n",
       "      'http://www.w3.org/2004/02/skos/core#prefLabel': 'NC_000020.11:g.34290848C>T (GRCh38)',\n",
       "      'https://terms.ga4gh.org/CanonicalReference': {'id': 'http://reg.genome.network/allele/CA408657898'}},\n",
       "     'alleleOrigin': {'id': 'obo:GENO_0000888'},\n",
       "     'zygosity': {'id': 'cg:TwoVariantsInTrans'},\n",
       "     'dc:source': {'id': 'https://pubmed.ncbi.nlm.nih.gov/31957987',\n",
       "      'type': 'dc:BibliographicResource',\n",
       "      'dc:abstract': 'S-adenosylhomocysteine hydrolase deficiency is an autosomal recessive neurometabolic disorder affecting the muscles, liver, and nervous system. The disease occurs by pathogenic variants of AHCY gene encoding S-adenosylhomocysteine hydrolase (AHCY) enzyme. This article reports a patient with presumed AHCY deficiency who was diagnosed by whole exome sequencing due to compound heterozygosity of novel p.T57I (c.170C>T) and p.V217M (c.649G>A) variants of AHCY gene. The patient had diffuse edema, coagulopathy, central nervous system abnormalities, and hypotonia. She died in 3\\u2009months due to cardiovascular collapse. Clinical findings of the present case were compatible with previously reported AHCY deficiency patients and the novel variants we found are considered to be the cause of the symptoms. This article also compiles the previous reports and expands clinical spectrum of AHCY deficiency by adding new features.',\n",
       "      'dc:creator': 'Bas H',\n",
       "      'dc:date': '2020',\n",
       "      'dc:title': 'A Turkish patient with novel AHCY variants and presumed diagnosis of S-adenosylhomocysteine hydrolase deficiency.'}},\n",
       "    {'id': 'cggv:109ef32f-7d2b-4ee3-b4b0-1de826fbfc12_variant_evidence_item',\n",
       "     'type': 'obo:SEPIO_0004116',\n",
       "     'allele': {'id': 'cggv:da6d5475-0147-4dc7-88a7-b14014c51784',\n",
       "      'type': 'https://terms.ga4gh.org/VariationDescriptor',\n",
       "      'http://www.w3.org/2004/02/skos/core#prefLabel': 'NC_000020.11:g.34295444G>A (GRCh38)',\n",
       "      'https://terms.ga4gh.org/CanonicalReference': {'id': 'http://reg.genome.network/allele/CA408660843'}},\n",
       "     'alleleOrigin': {'id': 'obo:GENO_0000888'},\n",
       "     'zygosity': {'id': 'cg:TwoVariantsInTrans'},\n",
       "     'dc:source': {'id': 'https://pubmed.ncbi.nlm.nih.gov/31957987',\n",
       "      'type': 'dc:BibliographicResource',\n",
       "      'dc:abstract': 'S-adenosylhomocysteine hydrolase deficiency is an autosomal recessive neurometabolic disorder affecting the muscles, liver, and nervous system. The disease occurs by pathogenic variants of AHCY gene encoding S-adenosylhomocysteine hydrolase (AHCY) enzyme. This article reports a patient with presumed AHCY deficiency who was diagnosed by whole exome sequencing due to compound heterozygosity of novel p.T57I (c.170C>T) and p.V217M (c.649G>A) variants of AHCY gene. The patient had diffuse edema, coagulopathy, central nervous system abnormalities, and hypotonia. She died in 3\\u2009months due to cardiovascular collapse. Clinical findings of the present case were compatible with previously reported AHCY deficiency patients and the novel variants we found are considered to be the cause of the symptoms. This article also compiles the previous reports and expands clinical spectrum of AHCY deficiency by adding new features.',\n",
       "      'dc:creator': 'Bas H',\n",
       "      'dc:date': '2020',\n",
       "      'dc:title': 'A Turkish patient with novel AHCY variants and presumed diagnosis of S-adenosylhomocysteine hydrolase deficiency.'}}],\n",
       "   'rdfs:label': 'Bas patient'},\n",
       "  {'id': 'cggv:788c32e5-57a0-48b9-8549-b2ecdbacef22',\n",
       "   'type': 'Proband',\n",
       "   'ageType': 'AgeAtReport',\n",
       "   'ageUnit': 'Months',\n",
       "   'ageValue': 2,\n",
       "   'variant': [{'id': 'cggv:8c54f7dd-e801-497f-ae03-11b7d59b3477_variant_evidence_item',\n",
       "     'type': 'obo:SEPIO_0004116',\n",
       "     'allele': {'id': 'cggv:f2cb1aab-a178-4938-9443-a47386171005'},\n",
       "     'alleleOrigin': {'id': 'obo:GENO_0000888'},\n",
       "     'zygosity': {'id': 'cg:TwoVariantsInTrans'},\n",
       "     'dc:source': {'id': 'https://pubmed.ncbi.nlm.nih.gov/27848944',\n",
       "      'type': 'dc:BibliographicResource',\n",
       "      'dc:abstract': 'We report our results of 1000 diagnostic WES cases based on 2819 sequenced samples from 54 countries with a wide phenotypic spectrum. Clinical information given by the requesting physicians was translated to HPO terms. WES processes were performed according to standardized settings. We identified the underlying pathogenic or likely pathogenic variants in 307 families (30.7%). In further 253 families (25.3%) a variant of unknown significance, possibly explaining the clinical symptoms of the index patient was identified. WES enabled timely diagnosing of genetic diseases, validation of causality of specific genetic disorders of PTPN23, KCTD3, SCN3A, PPOX, FRMPD4, and SCN1B, and setting dual diagnoses by detecting two causative variants in distinct genes in the same patient. We observed a better diagnostic yield in consanguineous families, in severe and in syndromic phenotypes. Our results suggest that WES has a better yield in patients that present with several symptoms, rather than an isolated abnormality. We also validate the clinical benefit of WES as an effective diagnostic tool, particularly in nonspecific or heterogeneous phenotypes. We recommend WES as a first-line diagnostic in all cases without a clear differential diagnosis, to facilitate personal medical care.',\n",
       "      'dc:creator': 'Trujillano D',\n",
       "      'dc:date': '2017',\n",
       "      'dc:title': 'Clinical exome sequencing: results from 2819 samples reflecting 1000 families.'}},\n",
       "    {'id': 'cggv:c0a9d02b-d48a-4ab2-8859-22ddb557e4b0_variant_evidence_item',\n",
       "     'type': 'obo:SEPIO_0004116',\n",
       "     'allele': {'id': 'cggv:79b7760a-855c-4ff4-a22e-49d891888cb3'},\n",
       "     'alleleOrigin': {'id': 'obo:GENO_0000888'},\n",
       "     'zygosity': {'id': 'cg:TwoVariantsInTrans'},\n",
       "     'dc:source': {'id': 'https://pubmed.ncbi.nlm.nih.gov/27848944',\n",
       "      'type': 'dc:BibliographicResource',\n",
       "      'dc:abstract': 'We report our results of 1000 diagnostic WES cases based on 2819 sequenced samples from 54 countries with a wide phenotypic spectrum. Clinical information given by the requesting physicians was translated to HPO terms. WES processes were performed according to standardized settings. We identified the underlying pathogenic or likely pathogenic variants in 307 families (30.7%). In further 253 families (25.3%) a variant of unknown significance, possibly explaining the clinical symptoms of the index patient was identified. WES enabled timely diagnosing of genetic diseases, validation of causality of specific genetic disorders of PTPN23, KCTD3, SCN3A, PPOX, FRMPD4, and SCN1B, and setting dual diagnoses by detecting two causative variants in distinct genes in the same patient. We observed a better diagnostic yield in consanguineous families, in severe and in syndromic phenotypes. Our results suggest that WES has a better yield in patients that present with several symptoms, rather than an isolated abnormality. We also validate the clinical benefit of WES as an effective diagnostic tool, particularly in nonspecific or heterogeneous phenotypes. We recommend WES as a first-line diagnostic in all cases without a clear differential diagnosis, to facilitate personal medical care.',\n",
       "      'dc:creator': 'Trujillano D',\n",
       "      'dc:date': '2017',\n",
       "      'dc:title': 'Clinical exome sequencing: results from 2819 samples reflecting 1000 families.'}}],\n",
       "   'rdfs:label': '00080802'},\n",
       "  {'id': 'cggv:98e0665d-940d-4efb-b547-940d52a22e24',\n",
       "   'type': 'Proband',\n",
       "   'ageType': 'AgeAtDeath',\n",
       "   'ageUnit': 'Months',\n",
       "   'ageValue': 11,\n",
       "   'allele': [{'id': 'cggv:80517393-27f0-42a7-a1e7-2ee4596f1fcb',\n",
       "     'type': 'https://terms.ga4gh.org/VariationDescriptor',\n",
       "     'http://www.w3.org/2004/02/skos/core#prefLabel': 'NC_000020.11:g.34295508G>A (GRCh38)',\n",
       "     'https://terms.ga4gh.org/CanonicalReference': {'id': 'http://reg.genome.network/allele/CA408660970'}},\n",
       "    {'id': 'cggv:bf1cc479-73a2-4502-b116-5978f58028a8',\n",
       "     'type': 'https://terms.ga4gh.org/VariationDescriptor',\n",
       "     'http://www.w3.org/2004/02/skos/core#prefLabel': 'NC_000020.11:g.34295472C>T (GRCh38)',\n",
       "     'https://terms.ga4gh.org/CanonicalReference': {'id': 'http://reg.genome.network/allele/CA408660902'}}],\n",
       "   'firstTestingMethod': 'Whole genome shotgun sequencing',\n",
       "   'phenotypeFreeText': 'plasma levels of methionine at 292 μmol/L (9–42 μmol/L), SAM at 5030 nM (33–95 nM), and SAH at 3290 nM (13–28 nM), CK of 1555 U/L (<355)',\n",
       "   'phenotypes': ['obo:HP_0002878',\n",
       "    'obo:HP_0012498',\n",
       "    'obo:HP_0001612',\n",
       "    'obo:HP_0001252',\n",
       "    'obo:HP_0002910',\n",
       "    'obo:HP_0003236',\n",
       "    'obo:HP_0001410',\n",
       "    'obo:HP_0003235',\n",
       "    'obo:HP_0003073',\n",
       "    'obo:HP_0001561',\n",
       "    'obo:HP_0030714',\n",
       "    'obo:HP_0002104'],\n",
       "   'sex': 'Male',\n",
       "   'variant': [{'id': 'cggv:39ec98c6-2ef0-4972-a043-02617a473186_variant_evidence_item',\n",
       "     'type': 'obo:SEPIO_0004116',\n",
       "     'allele': {'id': 'cggv:bf1cc479-73a2-4502-b116-5978f58028a8',\n",
       "      'type': 'https://terms.ga4gh.org/VariationDescriptor',\n",
       "      'http://www.w3.org/2004/02/skos/core#prefLabel': 'NC_000020.11:g.34295472C>T (GRCh38)',\n",
       "      'https://terms.ga4gh.org/CanonicalReference': {'id': 'http://reg.genome.network/allele/CA408660902'}},\n",
       "     'alleleOrigin': {'id': 'obo:GENO_0000877'},\n",
       "     'zygosity': {'id': 'cg:TwoVariantsInTrans'},\n",
       "     'dc:source': {'id': 'https://pubmed.ncbi.nlm.nih.gov/35789945',\n",
       "      'type': 'dc:BibliographicResource',\n",
       "      'dc:abstract': 'S-Adenosylhomocysteine (SAH) hydrolase deficiency is an autosomal recessive disorder in methionine metabolism caused by pathogenic variants in the gene ',\n",
       "      'dc:creator': 'Huang Y',\n",
       "      'dc:date': '2022',\n",
       "      'dc:title': 'The biochemical profile and dietary management in S-adenosylhomocysteine hydrolase deficiency.'}},\n",
       "    {'id': 'cggv:f56bd253-0e7c-4b28-af73-1b677a121a64_variant_evidence_item',\n",
       "     'type': 'obo:SEPIO_0004116',\n",
       "     'allele': {'id': 'cggv:80517393-27f0-42a7-a1e7-2ee4596f1fcb',\n",
       "      'type': 'https://terms.ga4gh.org/VariationDescriptor',\n",
       "      'http://www.w3.org/2004/02/skos/core#prefLabel': 'NC_000020.11:g.34295508G>A (GRCh38)',\n",
       "      'https://terms.ga4gh.org/CanonicalReference': {'id': 'http://reg.genome.network/allele/CA408660970'}},\n",
       "     'alleleOrigin': {'id': 'obo:GENO_0000877'},\n",
       "     'zygosity': {'id': 'cg:TwoVariantsInTrans'},\n",
       "     'dc:source': {'id': 'https://pubmed.ncbi.nlm.nih.gov/35789945',\n",
       "      'type': 'dc:BibliographicResource',\n",
       "      'dc:abstract': 'S-Adenosylhomocysteine (SAH) hydrolase deficiency is an autosomal recessive disorder in methionine metabolism caused by pathogenic variants in the gene ',\n",
       "      'dc:creator': 'Huang Y',\n",
       "      'dc:date': '2022',\n",
       "      'dc:title': 'The biochemical profile and dietary management in S-adenosylhomocysteine hydrolase deficiency.'}}],\n",
       "   'rdfs:label': 'Huang Proband'},\n",
       "  {'id': 'cggv:cf77708f-926f-4275-84fa-49fc7b4f894e',\n",
       "   'type': 'Proband',\n",
       "   'allele': [{'id': 'cggv:79b7760a-855c-4ff4-a22e-49d891888cb3'},\n",
       "    {'id': 'cggv:f2cb1aab-a178-4938-9443-a47386171005'}],\n",
       "   'sex': 'Male',\n",
       "   'variant': [{'id': 'cggv:d84cd273-0720-44c6-9cec-67dd878b434c_variant_evidence_item',\n",
       "     'type': 'obo:SEPIO_0004116',\n",
       "     'allele': {'id': 'cggv:79b7760a-855c-4ff4-a22e-49d891888cb3'},\n",
       "     'alleleOrigin': {'id': 'obo:GENO_0000880'},\n",
       "     'paternityMaternityConfirmed': 'No',\n",
       "     'zygosity': {'id': 'cg:TwoVariantsInTrans'},\n",
       "     'dc:source': {'id': 'https://pubmed.ncbi.nlm.nih.gov/16736098',\n",
       "      'type': 'dc:BibliographicResource',\n",
       "      'dc:abstract': \"This paper reports the third proven human case of deficient S-adenosylhomocysteine (AdoHcy) hydrolase activity. The patient is similar to the only two previously reported cases with this disorder in having severe myopathy, developmental delay, elevated serum creatine kinase (CK) concentrations, and hypermethioninaemia. Although he has been followed from infancy, the basic enzyme deficiency was established only at age 26 years. The diagnosis was based on markedly elevated plasma concentrations of both AdoHcy and S-adenosylmethionine, some 20% of the mean control activity of AdoHcy hydrolase activity in haemolysates of his red-blood cells, and two missense mutations in his gene encoding AdoHcy hydrolase. He had low values of erythrocyte phosphatidylcholine and plasma free choline and marginally elevated excretion of guanidinoacetate, suggesting that the elevated AdoHcy may have been inhibiting methylation of phosphatidylethanolamine and guanidinoacetate. His leukocyte DNA was globally more methylated than the DNA's of his parents or the mean extent of methylation measured in age-matched control subjects.\",\n",
       "      'dc:creator': 'Buist NR',\n",
       "      'dc:date': '2006',\n",
       "      'dc:title': 'S-adenosylhomocysteine hydrolase deficiency in a 26-year-old man.'}},\n",
       "    {'id': 'cggv:0fb58cb2-f500-484b-a91e-15cb473995d5_variant_evidence_item',\n",
       "     'type': 'obo:SEPIO_0004116',\n",
       "     'allele': {'id': 'cggv:f2cb1aab-a178-4938-9443-a47386171005'},\n",
       "     'alleleOrigin': {'id': 'obo:GENO_0000888'},\n",
       "     'zygosity': {'id': 'cg:TwoVariantsInTrans'},\n",
       "     'dc:source': {'id': 'https://pubmed.ncbi.nlm.nih.gov/16736098',\n",
       "      'type': 'dc:BibliographicResource',\n",
       "      'dc:abstract': \"This paper reports the third proven human case of deficient S-adenosylhomocysteine (AdoHcy) hydrolase activity. The patient is similar to the only two previously reported cases with this disorder in having severe myopathy, developmental delay, elevated serum creatine kinase (CK) concentrations, and hypermethioninaemia. Although he has been followed from infancy, the basic enzyme deficiency was established only at age 26 years. The diagnosis was based on markedly elevated plasma concentrations of both AdoHcy and S-adenosylmethionine, some 20% of the mean control activity of AdoHcy hydrolase activity in haemolysates of his red-blood cells, and two missense mutations in his gene encoding AdoHcy hydrolase. He had low values of erythrocyte phosphatidylcholine and plasma free choline and marginally elevated excretion of guanidinoacetate, suggesting that the elevated AdoHcy may have been inhibiting methylation of phosphatidylethanolamine and guanidinoacetate. His leukocyte DNA was globally more methylated than the DNA's of his parents or the mean extent of methylation measured in age-matched control subjects.\",\n",
       "      'dc:creator': 'Buist NR',\n",
       "      'dc:date': '2006',\n",
       "      'dc:title': 'S-adenosylhomocysteine hydrolase deficiency in a 26-year-old man.'}}],\n",
       "   'rdfs:label': 'Buist Case'}]}"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "jsonld.frame(input, frame)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "64bc4a39-a86a-4750-a052-33551fd1d2b0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
