{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "dc528bec-8c75-4851-8d31-fb355db617a7",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "#### Checks to be conducted :\n",
    "\n",
    "- Check for all the files where a proband misses dc:source or not and what can be done is such cases ?\n",
    "- Check whether the gene associated with the files is same as the gene associated with the variants ? But this needs not to be true\n",
    "- Check for the disease associated with the files and should we use them to fill **interpretations** element of the phenopacket, as it needs **diagnosis** element requires **disease**\n",
    "- For the phenotypes of a proband how many pubmed ids are associated with it ( theoritically 1 ) ?"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8df1a9b2-8c42-479f-b3f8-f6938be2a5e8",
   "metadata": {},
   "source": [
    "#### Issues pending\n",
    "- Error handling for certain dictionaries\n",
    "- Should we add gene ontology for allele descriptions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "df1f0ff8-5874-4f62-84e1-ffedbd6afebe",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Library Imports\n",
    "import csv\n",
    "import glob\n",
    "from google.protobuf.timestamp_pb2 import Timestamp\n",
    "import gzip\n",
    "import json\n",
    "import jsonpath_ng\n",
    "import mygene\n",
    "import phenopackets.schema.v2 as pps2\n",
    "import pyhpo\n",
    "import pyld"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e7bbd49d-6c1c-46c8-a088-098efe0895fc",
   "metadata": {},
   "source": [
    "#### Constants and housekeeping definitions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "387f3664-8d00-4536-a35f-42ce577c500c",
   "metadata": {},
   "outputs": [],
   "source": [
    "INPUT_FOLDER = '../../../Downloads/gene_disease_validity_20250616/'\n",
    "PUBMED_URL_ROOT = 'https://pubmed.ncbi.nlm.nih.gov/'\n",
    "timestamp = Timestamp()\n",
    "mg = mygene.MyGeneInfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7b9f7abd-4e60-4d87-8af1-c339ef07fb75",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dictionary MONDO_ID : label\n",
    "with gzip.open(\"MONDO.csv.gz\", \"rt\") as csvzip:\n",
    "    csvf = csv.DictReader(csvzip)\n",
    "    class_prefix = 'http://purl.obolibrary.org/obo/'\n",
    "    mondo_lookup = dict((row['Class ID'][len(class_prefix):],row['Preferred Label']) for row in csvf if row['Class ID'].startswith(class_prefix))\n",
    "\n",
    "# Test\n",
    "list(mondo_lookup.items())[:4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1af957a2-4f8e-48cb-b99c-f1b402a2be5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "pyhpo.Ontology()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9833674-34a6-470f-a6fb-8c7d1a1ef55f",
   "metadata": {},
   "source": [
    "#### Function Definitions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "38cbd686-54b0-4c32-b628-258bd1ed7c93",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check if the argument is list and if not convert it to list\n",
    "def ensure_list(value):\n",
    "    if isinstance(value,list):\n",
    "        return value\n",
    "    elif value == None:\n",
    "        return([])\n",
    "    else:\n",
    "        return([value])\n",
    "# Test\n",
    "ensure_list([])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fc160643-52c2-494f-ac37-103ee30480f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# check availability of the argument in the dicitionary and return the associated value, if not will return the argument\n",
    "def check_value(x, key):\n",
    "    if isinstance(x, dict):\n",
    "        return(x.get(key))\n",
    "    else:\n",
    "        return(key)\n",
    "\n",
    "# Test\n",
    "check_value(obo_to_labelled_phenotype ('obo:HP_0034519'), 'label') # Function defined later"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e48b247d-6ba2-4259-8acf-8f4cf415be44",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Test file for framing\n",
    "test_file = json.load(open(\"../../../Downloads/gene_disease_validity_20250616/cggv_0a0e2c4c-4d2f-4157-a611-75866f061bacv1.2.json\"))\n",
    "\n",
    "# JSON-LD Framing \n",
    "frame = {'@context': test_file['@context'], '@type': 'Proband', '@embed': '@always'}\n",
    "framed = pyld.jsonld.frame(test_file, frame)\n",
    "\n",
    "# Test \n",
    "framed['@graph'][0].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e335dbaf-65a6-4bd1-bbba-847baa0e4633",
   "metadata": {},
   "outputs": [],
   "source": [
    "def obo_to_labelled_phenotype(obo):\n",
    "    try:\n",
    "        term = pyhpo.Ontology.get_hpo_object(obo.replace('obo:HP_','HP:'))\n",
    "        return {'id' : term.id, 'label' : term.name}\n",
    "    except:\n",
    "        return {'id':obo}\n",
    "\n",
    "# Test\n",
    "obo_to_labelled_phenotype ('obo:HP_0034519') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ee39dd8-2428-4e7d-ba1e-50cfe4568596",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fall back to jsonpath_ng if proband doesn;t have 'dc:source' key\n",
    "dcsource_path = jsonpath_ng.parse(\"$..'dc:source'\") \n",
    "\n",
    "def find_sources(proband):\n",
    "    try:\n",
    "        dc_sources = [proband['dc:source']]\n",
    "    except KeyError:\n",
    "        try:\n",
    "            found = reversed(dcsource_path.find(proband))\n",
    "            sources_by_id = dict((x.value.get('id', 'UNKNOWN'), x.value)\n",
    "                                 for x in found\n",
    "                                 if not isinstance(x, str))\n",
    "            dc_sources = [sources_by_id.values()]\n",
    "        except Exception as e:\n",
    "            print(f\"Error finding dc:source with {proband=}\")\n",
    "            print(e)\n",
    "            dc_sources = []\n",
    "    return(dc_sources)\n",
    "\n",
    "# Test\n",
    "print(find_sources(framed[\"@graph\"][2]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3aa8ca56-90b5-47b3-b125-e8b8cc8314bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "def variants_extraction(proband):\n",
    "    variants = {}\n",
    "    try :\n",
    "        variants_data = proband['variant']\n",
    "        variants_data = ensure_list(variants_data)\n",
    "        for variant_number, item in enumerate(variants_data) :\n",
    "            # Might break if the variant in a proband doesn't have dc:source and assuming a variant has one dc:source in it\n",
    "            if isinstance(item[\"dc:source\"], dict):\n",
    "                pmid = 'PMID:' + item[\"dc:source\"][\"id\"].replace(PUBMED_URL_ROOT,'')\n",
    "            elif isinstance(item[\"dc:source\"], str):\n",
    "                pmid = 'PMID:' + item[\"dc:source\"].replace(PUBMED_URL_ROOT,'')\n",
    "            alleles = ensure_list(item[\"allele\"])\n",
    "            for allele_number, allele in enumerate(alleles):\n",
    "                variant = allele[\"http://www.w3.org/2004/02/skos/core#prefLabel\"]\n",
    "                variants[f'variant_{variant_number+1}.{allele_number+1}'] = {pmid:variant}\n",
    "    except Exception as e:\n",
    "        print(f'No variant found in {proband=}')\n",
    "    return variants\n",
    "\n",
    "# Should add .get method to counter the Keyerror associated with dc:source, allele, and http://www.w3.org/2004/02/skos/core#prefLabel keys    \n",
    "# Test\n",
    "variants_extraction(framed['@graph'][2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6c1b4fd-0a00-46da-966d-1323e2c27d58",
   "metadata": {},
   "outputs": [],
   "source": [
    "def proband_to_phenotype_report(proband):\n",
    "    phenotypes = [obo_to_labelled_phenotype(p) for p in ensure_list(proband.get('phenotypes',[]))]\n",
    "    sources = find_sources(proband)\n",
    "    try :\n",
    "        pmids = ','.join(set(s['id'].replace(PUBMED_URL_ROOT,'') for s in sources))\n",
    "    except:\n",
    "        pmids = ''\n",
    "    return { 'id' : proband['id'],\n",
    "             'pmids' : pmids,\n",
    "             'phenotypes' : phenotypes,\n",
    "             'label' : proband.get('rdfs:label',''),\n",
    "              'age' : str(proband.get('ageValue','')) + ' ' + proband.get('ageUnit','') + ' ' + proband.get('ageType',''),\n",
    "            'variants' : variants_extraction(proband)\n",
    "           }\n",
    "            \n",
    "\n",
    "# Test\n",
    "proband_to_phenotype_report(framed['@graph'][1])    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c63530b3-a464-490c-ab0c-4fc2c3356fa7",
   "metadata": {},
   "outputs": [],
   "source": [
    "files = sorted(glob.glob(f'{INPUT_FOLDER}/*.json'))\n",
    "len(list(files))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9e545f1a-f7a1-47cc-a353-3a5c1ce14ef1",
   "metadata": {},
   "outputs": [],
   "source": [
    "proband_summaries = {}\n",
    "for file in files:\n",
    "    with open(file, 'rt') as jf:\n",
    "        json_file = json.load(jf)\n",
    "        framed = pyld.jsonld.frame(json_file, frame)\n",
    "        for proband in framed.get('@graph',[]):\n",
    "            if proband['id'] in proband_summaries:\n",
    "                print(f'WARNING: more than one record for {proband[\"id\"]}, not sure which one to use.')\n",
    "            summary = proband_to_phenotype_report(proband)\n",
    "            summary['evidence_strength'] = json_file.get('evidenceStrength','')\n",
    "            proband_summaries[proband['id']] = summary        \n",
    "            "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "78f6829e-290d-4cf1-9aed-064f0cb30885",
   "metadata": {},
   "source": [
    "#### The structure of the data is not consistent\n",
    "\n",
    "- Sometimes the variants are single dictionaries other times they are list of dictionary key value pairs\n",
    "- Sometimes allelels in the variants are dictionary other times ther are list of dictionary key value pairs\n",
    "- Also, the dc:source of variants sometimes it has 'id' associated with it as a dictionary and other times there's only single element without id attribute"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "649f18c2-07f6-4c5b-91b2-4e6dbb547215",
   "metadata": {},
   "source": [
    "### Phenopacket Generation"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d167eec3-d0d1-454d-894c-0c5295a85d1c",
   "metadata": {},
   "source": [
    "#### Constants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "99b1e812-5206-4a77-9c8c-a94dfa80d04f",
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# Description of external resources used for referencing an object\n",
    "resources = [\n",
    "  pps2.Resource(**resource) for resource in [\n",
    "    {\n",
    "      \"id\": \"geno\",\n",
    "      \"name\": \"Genotype Ontology\",\n",
    "      \"url\": \"http://purl.obolibrary.org/obo/geno.owl\",\n",
    "      \"version\": \"2022-03-05\",\n",
    "      \"namespace_prefix\": \"GENO\",\n",
    "      \"iri_prefix\": \"http://purl.obolibrary.org/obo/GENO_\"\n",
    "    },\n",
    "    {\n",
    "      \"id\": \"hgnc\",\n",
    "      \"name\": \"HUGO Gene Nomenclature Committee\",\n",
    "      \"url\": \"https://www.genenames.org\",\n",
    "      \"version\": \"06/01/23\",\n",
    "      \"namespace_prefix\": \"HGNC\",\n",
    "      \"iri_prefix\": \"https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/\"\n",
    "    },\n",
    "    {\n",
    "      \"id\": \"omim\",\n",
    "      \"name\": \"An Online Catalog of Human Genes and Genetic Disorders\",\n",
    "      \"url\": \"https://www.omim.org\",\n",
    "      \"version\": \"January 4, 2023\",\n",
    "      \"namespace_prefix\": \"OMIM\",\n",
    "      \"iri_prefix\": \"https://www.omim.org/entry/\"\n",
    "    },\n",
    "    {\n",
    "      \"id\": \"hp\",\n",
    "      \"name\": \"human phenotype ontology\",\n",
    "      \"url\": \"http://purl.obolibrary.org/obo/hp.owl\",\n",
    "      \"version\": \"2024-07-01\",\n",
    "      \"namespace_prefix\": \"HP\",\n",
    "      \"iri_prefix\": \"http://purl.obolibrary.org/obo/HP_\"\n",
    "    },\n",
    "    { \"id\": \"mondo\",\n",
    "      \"name\": \"Mondo Disease Ontology\",\n",
    "      \"url\": \"http://purl.obolibrary.org/obo/mondo.owl\",\n",
    "      \"version\": \"2025-06-03\",\n",
    "      \"namespace_prefix\": \"MONDO\",\n",
    "      \"iri_prefix\": \"http://purl.obolibrary.org/obo/MONDO_\" \n",
    "    }\n",
    "  ]\n",
    "]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6e27a970-3e88-42fd-b2c6-f9373a2db663",
   "metadata": {},
   "source": [
    "#### Functions\n",
    "\n",
    "- Read the phenopacket documentation to understand the code, there is no api documentation avaialble while writing this code ( https://phenopacket-schema.readthedocs.io/en/latest/toplevel.html )\n",
    "- The level of heirarichies depends on the structure provided in the documentations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d5c33b1b-6603-4bdc-89dd-1a6bb2bbdd44",
   "metadata": {},
   "outputs": [],
   "source": [
    "def phenotype_element_from_id(hpoid, **kwargs):\n",
    "    term = pyhpo.Ontology.get_hpo_object(hpoid)\n",
    "    return pps2.PhenotypicFeature(type=pps2.OntologyClass(id=term.id, label=term.name), **kwargs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "40a54247-d33b-47be-a73d-59288e2c24f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "def disease_element_from_id( mondo_id ):\n",
    "    disease_id = mondo_id.replace('obo:','')\n",
    "    disease_label = mondo_lookup.get(disease_id,'')\n",
    "    disease = pps2.OntologyClass(id=disease_id.replace('_',':'), label = disease_label)\n",
    "    return(disease)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d94c2ecd-6704-4be2-8765-963d5de8b233",
   "metadata": {},
   "outputs": [],
   "source": [
    "def proband_to_phenopacket(proband, file):\n",
    "    references = []\n",
    "    referenceEvidence = []\n",
    "    for dc_sources in find_sources(proband):\n",
    "        for dc_source in dc_sources:\n",
    "            pmid = dc_source.get('id','').replace(PUBMED_URL_ROOT, '')\n",
    "            reference = pps2.ExternalReference(id=f'PMID:{pmid}', reference=dc_source['id'], description=dc_source['dc:title'])\n",
    "            referenceEvidence.append(pps2.Evidence(reference=reference, evidence_code=pps2.OntologyClass(id=\"ECO:0006017\", label=\"author statement from published clinical study used in manual assertion\")))\n",
    "            references.append(reference)\n",
    "    phenotypes = [phenotype_element_from_id(p.replace('obo:HP_', 'HP:'), evidence=referenceEvidence) for p in proband.get('phenotypes', [])]\n",
    "    metadata = pps2.MetaData(external_references=references,\n",
    "                            created = timestamp.GetCurrentTime(),\n",
    "                            created_by=\"Automated import from ClinGen GCI data\",\n",
    "                            phenopacket_schema_version=\"2.0\",\n",
    "                             resources = resources\n",
    "                            )\n",
    "    subject_args = {'id': f'PMID_{pmid}:{proband['rdfs:label']}'}\n",
    "    subject_args['sex'] = pps2.Sex.Value(proband.get('sex','UNKNOWN').upper())\n",
    "    individual = pps2.Individual(**subject_args)   \n",
    "    for key, value in variants_extraction(proband).items():\n",
    "        genomic_interpretations = []\n",
    "        for k,v in value.items():\n",
    "            variant_interpretation = (pps2.VariantInterpretation( acmg_pathogenicity_classification = 'NOT_PROVIDED',\n",
    "                                          variation_descriptor = pps2.VariationDescriptor(id=  k + '_' + key,\n",
    "                               expressions = [pps2.Expression(syntax = 'hgvs', value = v)])))\n",
    "            genomic_interpretations.append(pps2.GenomicInterpretation(subject_or_biosample_id = proband['id'],\n",
    "                                         interpretation_status = 'UNKNOWN_STATUS',\n",
    "                                         variant_interpretation = variant_interpretation ))\n",
    "    diagnosis = pps2.Diagnosis( disease = disease_element_from_id(file['subject']['disease']),\n",
    "               genomic_interpretations = genomic_interpretations)\n",
    "    interpretatioin = pps2.Interpretation( id = proband['id'],\n",
    "                     progress_status = 'UNKNOWN_PROGRESS',\n",
    "                    diagnosis = diagnosis\n",
    "                   )\n",
    "        \n",
    "        \n",
    "    phenopacket = pps2.Phenopacket(id=proband['id'], subject=individual, phenotypic_features=phenotypes, meta_data=metadata, interpretations = [interpretatioin])\n",
    "    return phenopacket\n",
    "\n",
    "# Test\n",
    "proband_to_phenopacket(proband,test_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e0e4b9ce-0285-424e-b536-1daa8f9eb749",
   "metadata": {},
   "outputs": [],
   "source": [
    "proband"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "35b827fe-8067-46a2-a3c2-3c2b2291d791",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "### Scratch Code for testing"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "76d67de5-42a7-4ffb-9f62-6c2c39114c15",
   "metadata": {},
   "source": [
    "#### Subject "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ee6bd1a-e90d-4e7c-8417-1f05c0cf7c69",
   "metadata": {},
   "outputs": [],
   "source": [
    "pps2.Individual(id= str(proband.get('dc:source','')['id'].replace('https://pubmed.ncbi.nlm.nih.gov/','')) + ':'+ str(proband.get('rdfs:label','')), sex=proband.get(\"sex\",\"UNKNOWN\").upper())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fcdade7a-8c8f-4039-8980-e5929d1a5661",
   "metadata": {},
   "source": [
    "#### Phenotypic Feature"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e99deb4-5e99-4947-85c2-75d49652828a",
   "metadata": {},
   "outputs": [],
   "source": [
    "pps2.PhenotypicFeature(type = pps2.OntologyClass(id='HP:0004758', label='Effort-induced polymorphic ventricular tachycardia'),\n",
    "                      evidence=[pps2.Evidence(evidence_code = pps2.OntologyClass(id='ECO:0006017', label='author statement supported from referenced clinical study used in manual assertion'),\n",
    "             reference =pps2.ExternalReference(id=\"PMID:15178757\", reference='https://pubmed.ncbi.nlm.nih.gov/15178757',\n",
    "                       description='A cardiac arrhythmia syndrome caused by loss of ankyrin-B function.'))])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fdb3fbfe-8f32-4610-81e0-88aad6cc8d04",
   "metadata": {},
   "source": [
    "#### Ontology Class"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "11b998c1-9299-44c4-b906-30ff40fdf0c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "pps2.OntologyClass(id='HP:0004758', label='Effort-induced polymorphic ventricular tachycardia')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e245c2d-c9ac-4a98-8529-3f2575ff5b83",
   "metadata": {},
   "source": [
    "#### Evidence "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f35d5a6-f691-44b8-baf2-bbbe5066cb04",
   "metadata": {},
   "outputs": [],
   "source": [
    "pps2.Evidence(evidence_code = pps2.OntologyClass(id='ECO:0006017', label='author statement supported from referenced clinical study used in manual assertion'),\n",
    "             reference =pps2.ExternalReference(id=\"PMID:15178757\", reference='https://pubmed.ncbi.nlm.nih.gov/15178757',\n",
    "                       description='A cardiac arrhythmia syndrome caused by loss of ankyrin-B function.') )"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2a1771f8-229e-482b-b2ad-e27b7daebcce",
   "metadata": {},
   "source": [
    "##### External ReferenceÂ¶"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ecf3517d-79e1-4c8c-9b3c-5db4be9dc3e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "pps2.ExternalReference(id=\"PMID:15178757\", reference='https://pubmed.ncbi.nlm.nih.gov/15178757',\n",
    "                       description='A cardiac arrhythmia syndrome caused by loss of ankyrin-B function.')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb7ed8c9-a06d-44dd-bfcd-4096803598f0",
   "metadata": {},
   "source": [
    "#### Metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "318fa215-0ec9-4498-bfe5-63849e08668b",
   "metadata": {},
   "outputs": [],
   "source": [
    "pps2.MetaData(\n",
    "    created = timestamp.GetCurrentTime(),\n",
    "    created_by = \"Automated import from ClinGen GCI data\",\n",
    "    phenopacket_schema_version = \"2.0\",\n",
    "    resources = resources,\n",
    " external_references = [pps2.ExternalReference(id=\"PMID:15178757\", reference='https://pubmed.ncbi.nlm.nih.gov/15178757',\n",
    "                       description='A cardiac arrhythmia syndrome caused by loss of ankyrin-B function.')]\n",
    ")\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8eb50740-cf3b-460b-a45e-9617b36a1f7d",
   "metadata": {},
   "outputs": [],
   "source": [
    "pps2.Interpretation(id='123456',\n",
    "                    progress_status='UNKNOWN_PROGRESS',\n",
    "                    diagnosis = pps2.Diagnosis( disease = pps2.OntologyClass(id='MONDO:0044647',label='kyphosis-lateral tongue atrophy-myofibrillar myopathy syndrome'),\n",
    "               genomic_interpretations = genomic_interpretations))\n",
    "\n",
    "                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8507d574-1c79-446e-aaa2-1c1dbd13b038",
   "metadata": {},
   "outputs": [],
   "source": [
    "pps2.Diagnosis( disease = pps2.OntologyClass(id='MONDO:0044647',label='kyphosis-lateral tongue atrophy-myofibrillar myopathy syndrome'),\n",
    "               genomic_interpretations = genomic_interpretations)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9fdfb42f-8013-465e-aeb0-abc676ebcae4",
   "metadata": {},
   "outputs": [],
   "source": [
    "pps2.Disease(pps2.OntologyClass(id='MONDO:0044647',label='kyphosis-lateral tongue atrophy-myofibrillar myopathy syndrome'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "829ff5c3-37dc-4558-9877-f15c86c31565",
   "metadata": {},
   "outputs": [],
   "source": [
    "genomic_interpretations = [pps2.GenomicInterpretation( subject_or_biosample_id = 'AJBKS',\n",
    "                            interpretation_status = 'UNKNOWN_STATUS',\n",
    "                            variant_interpretation = pps2.VariantInterpretation( acmg_pathogenicity_classification = 'NOT_PROVIDED',\n",
    "                           variation_descriptor = pps2.VariationDescriptor(id='12344',\n",
    "                               expressions = [pps2.Expression(syntax = 'hgvs', value = 'NM_172056.2(KCNH2):c.1764C>A (p.Asn588Lys)')],\n",
    "                                                                           \n",
    "                                                                          )))]\n",
    "genomic_interpretations                          "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f18d1ab9-9a42-494e-892e-284108ec740d",
   "metadata": {},
   "outputs": [],
   "source": [
    "pps2.VariantInterpretation( acmg_pathogenicity_classification = 'NOT_PROVIDED',\n",
    "                           variation_descriptor = pps2.VariationDescriptor(id='12344',\n",
    "                               expressions = [pps2.Expression(syntax = 'hgvs', value = 'NM_172056.2(KCNH2):c.1764C>A (p.Asn588Lys)')],\n",
    "                                                                           \n",
    "                                                                          )\n",
    "                          )\n",
    "                           "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e57b32a-dd19-47b2-857e-c9398da1e7b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "pps2.Expression(syntax = 'hgvs', value = 'NM_172056.2(KCNH2):c.1764C>A (p.Asn588Lys)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5939857b-8b8c-42f8-bd99-8c1a7964ebdc",
   "metadata": {},
   "outputs": [],
   "source": [
    "pps2.OntologyClass(id = 'GENO_0000888', label = 'germline allele origin')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "04aa9075-f73d-49f8-bc95-99d68410de98",
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6cbf299c-9650-4792-90f4-e89612d81b1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "requests.get('http://purl.obolibrary.org/obo/','GENO_0000888')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
