{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "68776297-c56e-47b1-9c72-65fdf28f938b",
   "metadata": {},
   "source": [
    "##### A quick notebook utilizing pandas to get quick stats out of the genegraph data / scratch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2000346e-ce53-4faa-bf96-82fee38f206a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Library Imports\n",
    "import glob\n",
    "import json\n",
    "import jsonpath_ng\n",
    "import os \n",
    "import pandas as pd\n",
    "from pyld import jsonld\n",
    "import "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8405eacd-03aa-4d6f-adc0-be9764703db4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Constants\n",
    "INPUT_FOLDER = './input/gene-validity-jsonld-20251030/gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json'\n",
    "MONDO_MAPPINGS = './input/MONDO_20251030.csv.gz'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "78a695ed-6856-4051-ae17-b7a5d42388a7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/06/r5yblj5x0r30hcgx835svxkm0000gn/T/ipykernel_23020/1278791242.py:2: DtypeWarning: Columns (14,104) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  mondo_mappings = pd.read_csv(MONDO_MAPPINGS, compression='gzip')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class ID</th>\n",
       "      <th>Preferred Label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MONDO_0044647</td>\n",
       "      <td>kyphosis-lateral tongue atrophy-myofibrillar m...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MONDO_1010032</td>\n",
       "      <td>Jacobsen syndrome, non-human animal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MONDO_0007146</td>\n",
       "      <td>obsolete apnea, central sleep</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MONDO_0017964</td>\n",
       "      <td>obsolete 46,XX disorder of sex development ind...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MONDO_0014635</td>\n",
       "      <td>microphthalmia, isolated, with coloboma 10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Class ID                                    Preferred Label\n",
       "0  MONDO_0044647  kyphosis-lateral tongue atrophy-myofibrillar m...\n",
       "1  MONDO_1010032                Jacobsen syndrome, non-human animal\n",
       "2  MONDO_0007146                      obsolete apnea, central sleep\n",
       "3  MONDO_0017964  obsolete 46,XX disorder of sex development ind...\n",
       "4  MONDO_0014635         microphthalmia, isolated, with coloboma 10"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# mondo lookup to map ids to names\n",
    "mondo_mappings = pd.read_csv(MONDO_MAPPINGS, compression='gzip')\n",
    "mondo_lookup = mondo_mappings.loc[:,['Class ID', 'Preferred Label']]\n",
    "mondo_lookup['Class ID'] = mondo_lookup['Class ID'].map( lambda x : x.strip('http://purl.obolibrary.org/obo/'))\n",
    "mondo_lookup.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "87c64c9f-b552-4fae-8015-1a8de168ec58",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'obo:MONDO_0700068'"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Reading a file to check what it has \n",
    "genegraph_validity_jsonld_sample = json.load(open(INPUT_FOLDER + '/gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json'))\n",
    "genegraph_validity_jsonld_sample['subject']['disease']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5f0acd50-bf46-4899-9e85-6ca01632d260",
   "metadata": {},
   "source": [
    "- Used json_ld for further processing\n",
    "- Trying to use @embed : always [ assumption it will resolve the labels to proper uris in the files ] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "90b3101d-df6e-4af7-bf31-8c8f69b774a2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'id': 'https://genegraph.clinicalgenome.org/r/0708860f-3679-4c7c-8809-631729d6490a',\n",
       " 'type': 'Proband',\n",
       " 'dc:source': 'https://pubmed.ncbi.nlm.nih.gov/11709191',\n",
       " 'rdfs:label': 'Yoshida_Patient MK',\n",
       " 'ageType': 'AgeAtReport',\n",
       " 'ageUnit': 'Years',\n",
       " 'ageValue': 8,\n",
       " 'allele': {'id': 'https://genegraph.clinicalgenome.org/r/9b9f6f3a-715d-4fcf-9271-10d1cd54cb8f',\n",
       "  'type': 'https://terms.ga4gh.org/VariationDescriptor',\n",
       "  'http://www.w3.org/2004/02/skos/core#prefLabel': 'NM_017739.3(POMGNT1):c.1649G>A (p.Ser550Asn)',\n",
       "  'https://terms.ga4gh.org/CanonicalReference': {'id': 'http://reg.genome.network/allele/CA116540'}},\n",
       " 'detectionMethod': 'The entire coding region (ex 1-22) and intron-exon flanking sequences of POMGNT1 were amplified and directly sequenced.',\n",
       " 'firstTestingMethod': 'PCR',\n",
       " 'phenotypeFreeText': 'Clinical features from PMID: 12588800: Moderate white matter lucency. Proband did not show head control at 8yo. CK levels at 4yo = 628 U/L.',\n",
       " 'phenotypes': ['obo:HP_0007973',\n",
       "  'obo:HP_0007260',\n",
       "  'obo:HP_0001320',\n",
       "  'obo:HP_0002079',\n",
       "  'obo:HP_0002365',\n",
       "  'obo:HP_0010864'],\n",
       " 'previousTesting': True,\n",
       " 'previousTestingDescription': 'Linkage analysis was performed to narrow down the MEB locus in families by founder haplotype mapping.',\n",
       " 'secondTestingMethod': 'Sanger sequencing',\n",
       " 'sex': 'Male',\n",
       " 'variant': {'id': 'https://genegraph.clinicalgenome.org/r/23dbada2-4f87-46f1-94d4-c5befd832b58_variant_evidence_item',\n",
       "  'type': 'VariantObservation',\n",
       "  'dc:source': 'https://pubmed.ncbi.nlm.nih.gov/11709191',\n",
       "  'allele': {'id': 'https://genegraph.clinicalgenome.org/r/9b9f6f3a-715d-4fcf-9271-10d1cd54cb8f',\n",
       "   'type': 'https://terms.ga4gh.org/VariationDescriptor',\n",
       "   'http://www.w3.org/2004/02/skos/core#prefLabel': 'NM_017739.3(POMGNT1):c.1649G>A (p.Ser550Asn)',\n",
       "   'https://terms.ga4gh.org/CanonicalReference': {'id': 'http://reg.genome.network/allele/CA116540'}},\n",
       "  'alleleOrigin': {'id': 'cg:GermlineAlleleOrigin'},\n",
       "  'zygosity': {'id': 'cg:Homozygous'}}}"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Getting a frame to extract all the entities associated with proband data along with the gene and disease\n",
    "frame = {'@context': genegraph_validity_jsonld_sample['@context'], '@type': 'Proband', '@embed': '@always'}\n",
    "framed_data = jsonld.frame(genegraph_validity_jsonld_sample, frame)\n",
    "framed_data[\"@graph\"][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e179ca1a-c18e-4246-83a6-a89e98868bcf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>type</th>\n",
       "      <th>dc:source</th>\n",
       "      <th>rdfs:label</th>\n",
       "      <th>ageType</th>\n",
       "      <th>ageUnit</th>\n",
       "      <th>ageValue</th>\n",
       "      <th>allele</th>\n",
       "      <th>detectionMethod</th>\n",
       "      <th>firstTestingMethod</th>\n",
       "      <th>phenotypeFreeText</th>\n",
       "      <th>phenotypes</th>\n",
       "      <th>previousTesting</th>\n",
       "      <th>previousTestingDescription</th>\n",
       "      <th>secondTestingMethod</th>\n",
       "      <th>sex</th>\n",
       "      <th>variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/0708860...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/11709191</td>\n",
       "      <td>Yoshida_Patient MK</td>\n",
       "      <td>AgeAtReport</td>\n",
       "      <td>Years</td>\n",
       "      <td>8</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>The entire coding region (ex 1-22) and intron-...</td>\n",
       "      <td>PCR</td>\n",
       "      <td>Clinical features from PMID: 12588800: Moderat...</td>\n",
       "      <td>[obo:HP_0007973, obo:HP_0007260, obo:HP_000132...</td>\n",
       "      <td>True</td>\n",
       "      <td>Linkage analysis was performed to narrow down ...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Male</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  id     type  \\\n",
       "0  https://genegraph.clinicalgenome.org/r/0708860...  Proband   \n",
       "\n",
       "                                  dc:source          rdfs:label      ageType  \\\n",
       "0  https://pubmed.ncbi.nlm.nih.gov/11709191  Yoshida_Patient MK  AgeAtReport   \n",
       "\n",
       "  ageUnit  ageValue                                             allele  \\\n",
       "0   Years         8  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "\n",
       "                                     detectionMethod firstTestingMethod  \\\n",
       "0  The entire coding region (ex 1-22) and intron-...                PCR   \n",
       "\n",
       "                                   phenotypeFreeText  \\\n",
       "0  Clinical features from PMID: 12588800: Moderat...   \n",
       "\n",
       "                                          phenotypes  previousTesting  \\\n",
       "0  [obo:HP_0007973, obo:HP_0007260, obo:HP_000132...             True   \n",
       "\n",
       "                          previousTestingDescription secondTestingMethod  \\\n",
       "0  Linkage analysis was performed to narrow down ...   Sanger sequencing   \n",
       "\n",
       "    sex                                            variant  \n",
       "0  Male  {'id': 'https://genegraph.clinicalgenome.org/r...  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame.from_dict([framed_data[\"@graph\"][0]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "8c259cf0-82b5-42b7-9a4a-cc6752e8dfb4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3371"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Directory \n",
    "genegraph_json_files_path = sorted(glob.glob(f'{INPUT_FOLDER}/*.json'))\n",
    "len(genegraph_json_files_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "4cba6451-6299-4fd4-bce7-4ea2fdfa2691",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "An exception Length of values (2) does not match length of index (1) occur in reading file at ./input/gene-validity-jsonld-20251030/gg_107f9b2d-8bc2-4165-9017-78c78b8151e0v1.0.json\n",
      "An exception Length of values (2) does not match length of index (1) occur in reading file at ./input/gene-validity-jsonld-20251030/gg_7c114417-af28-41d9-8434-f99db9a16543v1.0.json\n",
      "An exception 'subject' occur in reading file at ./input/gene-validity-jsonld-20251030/gg_87a36506-8821-4ddc-8ab8-6fa97936ee02.json\n"
     ]
    }
   ],
   "source": [
    "# Creating a common dataframe with all the probands data\n",
    "\n",
    "all_proband_data = [] \n",
    "\n",
    "for file_path in genegraph_json_files_path:\n",
    "    try:\n",
    "        with open(file_path) as jf:\n",
    "            file_content = json.load(jf)\n",
    "            gene = file_content['subject']['gene'] \n",
    "            disease = file_content['subject']['disease'] \n",
    "            framed_data = jsonld.frame(file_content, frame)\n",
    "            for proband in framed_data.get('@graph',[]):\n",
    "                temp_proband_df = pd.DataFrame.from_dict([proband])\n",
    "                temp_proband_df[\"gene\"] = gene\n",
    "                temp_proband_df[\"disease\"]= disease\n",
    "                temp_proband_df[\"source_file\"] = os.path.basename(file_path)\n",
    "                all_proband_data.append(temp_proband_df)\n",
    "    except Exception as e:\n",
    "        print(f\"An exception {e} occur in reading file at {file_path}\")\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "7fc31754-3357-463b-a8a8-0ed6f05030f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "proband_data = pd.concat(all_proband_data, ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "8393b5e0-2bbf-4536-8c53-3e20d27b7d02",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>type</th>\n",
       "      <th>dc:source</th>\n",
       "      <th>rdfs:label</th>\n",
       "      <th>ageType</th>\n",
       "      <th>ageUnit</th>\n",
       "      <th>ageValue</th>\n",
       "      <th>allele</th>\n",
       "      <th>detectionMethod</th>\n",
       "      <th>firstTestingMethod</th>\n",
       "      <th>...</th>\n",
       "      <th>previousTesting</th>\n",
       "      <th>previousTestingDescription</th>\n",
       "      <th>secondTestingMethod</th>\n",
       "      <th>sex</th>\n",
       "      <th>variant</th>\n",
       "      <th>gene</th>\n",
       "      <th>disease</th>\n",
       "      <th>source_file</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/0708860...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/11709191</td>\n",
       "      <td>Yoshida_Patient MK</td>\n",
       "      <td>AgeAtReport</td>\n",
       "      <td>Years</td>\n",
       "      <td>8.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>The entire coding region (ex 1-22) and intron-...</td>\n",
       "      <td>PCR</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>Linkage analysis was performed to narrow down ...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Male</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>obo:MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/352d557...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/26908613</td>\n",
       "      <td>Xu_Proband 2</td>\n",
       "      <td>AgeAtReport</td>\n",
       "      <td>Years</td>\n",
       "      <td>32.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>Reanalysis of WES data revealed the POMGNT1 va...</td>\n",
       "      <td>Genotyping</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>WES was performed but the cause of RP was unid...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Female</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>obo:MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/5f3835c...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/11709191</td>\n",
       "      <td>Yoshida_Patient SA</td>\n",
       "      <td>AgeAtReport</td>\n",
       "      <td>Years</td>\n",
       "      <td>7.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>The entire coding region (ex 1-22) and intron-...</td>\n",
       "      <td>PCR</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>Linkage analysis was performed to narrow down ...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Female</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>obo:MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/612230d...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/11709191</td>\n",
       "      <td>Yoshida_Patient YA</td>\n",
       "      <td>AgeAtReport</td>\n",
       "      <td>Years</td>\n",
       "      <td>6.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>The entire coding region (ex 1-22) and intron-...</td>\n",
       "      <td>PCR</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>Linkage analysis was performed to narrow down ...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Male</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>obo:MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/69b2ace...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/18195152</td>\n",
       "      <td>Clement_Proband</td>\n",
       "      <td>AgeAtOnset</td>\n",
       "      <td>Years</td>\n",
       "      <td>12.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>The POMGNT1 gene was screened for mutations.</td>\n",
       "      <td>Other</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>Abnormal glycosylation of α-DG was noted on mu...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Female</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>obo:MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  id     type  \\\n",
       "0  https://genegraph.clinicalgenome.org/r/0708860...  Proband   \n",
       "1  https://genegraph.clinicalgenome.org/r/352d557...  Proband   \n",
       "2  https://genegraph.clinicalgenome.org/r/5f3835c...  Proband   \n",
       "3  https://genegraph.clinicalgenome.org/r/612230d...  Proband   \n",
       "4  https://genegraph.clinicalgenome.org/r/69b2ace...  Proband   \n",
       "\n",
       "                                  dc:source          rdfs:label      ageType  \\\n",
       "0  https://pubmed.ncbi.nlm.nih.gov/11709191  Yoshida_Patient MK  AgeAtReport   \n",
       "1  https://pubmed.ncbi.nlm.nih.gov/26908613        Xu_Proband 2  AgeAtReport   \n",
       "2  https://pubmed.ncbi.nlm.nih.gov/11709191  Yoshida_Patient SA  AgeAtReport   \n",
       "3  https://pubmed.ncbi.nlm.nih.gov/11709191  Yoshida_Patient YA  AgeAtReport   \n",
       "4  https://pubmed.ncbi.nlm.nih.gov/18195152     Clement_Proband   AgeAtOnset   \n",
       "\n",
       "  ageUnit  ageValue                                             allele  \\\n",
       "0   Years       8.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "1   Years      32.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "2   Years       7.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "3   Years       6.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "4   Years      12.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "\n",
       "                                     detectionMethod firstTestingMethod  ...  \\\n",
       "0  The entire coding region (ex 1-22) and intron-...                PCR  ...   \n",
       "1  Reanalysis of WES data revealed the POMGNT1 va...         Genotyping  ...   \n",
       "2  The entire coding region (ex 1-22) and intron-...                PCR  ...   \n",
       "3  The entire coding region (ex 1-22) and intron-...                PCR  ...   \n",
       "4       The POMGNT1 gene was screened for mutations.              Other  ...   \n",
       "\n",
       "  previousTesting                         previousTestingDescription  \\\n",
       "0            True  Linkage analysis was performed to narrow down ...   \n",
       "1            True  WES was performed but the cause of RP was unid...   \n",
       "2            True  Linkage analysis was performed to narrow down ...   \n",
       "3            True  Linkage analysis was performed to narrow down ...   \n",
       "4            True  Abnormal glycosylation of α-DG was noted on mu...   \n",
       "\n",
       "  secondTestingMethod     sex  \\\n",
       "0   Sanger sequencing    Male   \n",
       "1   Sanger sequencing  Female   \n",
       "2   Sanger sequencing  Female   \n",
       "3   Sanger sequencing    Male   \n",
       "4   Sanger sequencing  Female   \n",
       "\n",
       "                                             variant        gene  \\\n",
       "0  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "1  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "2  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "3  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "4  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "\n",
       "             disease                                       source_file  \\\n",
       "0  obo:MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json   \n",
       "1  obo:MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json   \n",
       "2  obo:MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json   \n",
       "3  obo:MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json   \n",
       "4  obo:MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json   \n",
       "\n",
       "  ethnicity zygosity  \n",
       "0       NaN      NaN  \n",
       "1       NaN      NaN  \n",
       "2       NaN      NaN  \n",
       "3       NaN      NaN  \n",
       "4       NaN      NaN  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proband_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "c986e6ac-b699-4d43-87d2-c774bb84a402",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>type</th>\n",
       "      <th>dc:source</th>\n",
       "      <th>rdfs:label</th>\n",
       "      <th>ageType</th>\n",
       "      <th>ageUnit</th>\n",
       "      <th>ageValue</th>\n",
       "      <th>allele</th>\n",
       "      <th>detectionMethod</th>\n",
       "      <th>firstTestingMethod</th>\n",
       "      <th>...</th>\n",
       "      <th>previousTesting</th>\n",
       "      <th>previousTestingDescription</th>\n",
       "      <th>secondTestingMethod</th>\n",
       "      <th>sex</th>\n",
       "      <th>variant</th>\n",
       "      <th>gene</th>\n",
       "      <th>disease</th>\n",
       "      <th>source_file</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/0708860...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/11709191</td>\n",
       "      <td>Yoshida_Patient MK</td>\n",
       "      <td>AgeAtReport</td>\n",
       "      <td>Years</td>\n",
       "      <td>8.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>The entire coding region (ex 1-22) and intron-...</td>\n",
       "      <td>PCR</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>Linkage analysis was performed to narrow down ...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Male</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/352d557...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/26908613</td>\n",
       "      <td>Xu_Proband 2</td>\n",
       "      <td>AgeAtReport</td>\n",
       "      <td>Years</td>\n",
       "      <td>32.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>Reanalysis of WES data revealed the POMGNT1 va...</td>\n",
       "      <td>Genotyping</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>WES was performed but the cause of RP was unid...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Female</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/5f3835c...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/11709191</td>\n",
       "      <td>Yoshida_Patient SA</td>\n",
       "      <td>AgeAtReport</td>\n",
       "      <td>Years</td>\n",
       "      <td>7.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>The entire coding region (ex 1-22) and intron-...</td>\n",
       "      <td>PCR</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>Linkage analysis was performed to narrow down ...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Female</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/612230d...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/11709191</td>\n",
       "      <td>Yoshida_Patient YA</td>\n",
       "      <td>AgeAtReport</td>\n",
       "      <td>Years</td>\n",
       "      <td>6.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>The entire coding region (ex 1-22) and intron-...</td>\n",
       "      <td>PCR</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>Linkage analysis was performed to narrow down ...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Male</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/69b2ace...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/18195152</td>\n",
       "      <td>Clement_Proband</td>\n",
       "      <td>AgeAtOnset</td>\n",
       "      <td>Years</td>\n",
       "      <td>12.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>The POMGNT1 gene was screened for mutations.</td>\n",
       "      <td>Other</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>Abnormal glycosylation of α-DG was noted on mu...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Female</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  id     type  \\\n",
       "0  https://genegraph.clinicalgenome.org/r/0708860...  Proband   \n",
       "1  https://genegraph.clinicalgenome.org/r/352d557...  Proband   \n",
       "2  https://genegraph.clinicalgenome.org/r/5f3835c...  Proband   \n",
       "3  https://genegraph.clinicalgenome.org/r/612230d...  Proband   \n",
       "4  https://genegraph.clinicalgenome.org/r/69b2ace...  Proband   \n",
       "\n",
       "                                  dc:source          rdfs:label      ageType  \\\n",
       "0  https://pubmed.ncbi.nlm.nih.gov/11709191  Yoshida_Patient MK  AgeAtReport   \n",
       "1  https://pubmed.ncbi.nlm.nih.gov/26908613        Xu_Proband 2  AgeAtReport   \n",
       "2  https://pubmed.ncbi.nlm.nih.gov/11709191  Yoshida_Patient SA  AgeAtReport   \n",
       "3  https://pubmed.ncbi.nlm.nih.gov/11709191  Yoshida_Patient YA  AgeAtReport   \n",
       "4  https://pubmed.ncbi.nlm.nih.gov/18195152     Clement_Proband   AgeAtOnset   \n",
       "\n",
       "  ageUnit  ageValue                                             allele  \\\n",
       "0   Years       8.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "1   Years      32.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "2   Years       7.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "3   Years       6.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "4   Years      12.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "\n",
       "                                     detectionMethod firstTestingMethod  ...  \\\n",
       "0  The entire coding region (ex 1-22) and intron-...                PCR  ...   \n",
       "1  Reanalysis of WES data revealed the POMGNT1 va...         Genotyping  ...   \n",
       "2  The entire coding region (ex 1-22) and intron-...                PCR  ...   \n",
       "3  The entire coding region (ex 1-22) and intron-...                PCR  ...   \n",
       "4       The POMGNT1 gene was screened for mutations.              Other  ...   \n",
       "\n",
       "  previousTesting                         previousTestingDescription  \\\n",
       "0            True  Linkage analysis was performed to narrow down ...   \n",
       "1            True  WES was performed but the cause of RP was unid...   \n",
       "2            True  Linkage analysis was performed to narrow down ...   \n",
       "3            True  Linkage analysis was performed to narrow down ...   \n",
       "4            True  Abnormal glycosylation of α-DG was noted on mu...   \n",
       "\n",
       "  secondTestingMethod     sex  \\\n",
       "0   Sanger sequencing    Male   \n",
       "1   Sanger sequencing  Female   \n",
       "2   Sanger sequencing  Female   \n",
       "3   Sanger sequencing    Male   \n",
       "4   Sanger sequencing  Female   \n",
       "\n",
       "                                             variant        gene  \\\n",
       "0  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "1  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "2  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "3  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "4  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "\n",
       "         disease                                       source_file ethnicity  \\\n",
       "0  MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json       NaN   \n",
       "1  MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json       NaN   \n",
       "2  MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json       NaN   \n",
       "3  MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json       NaN   \n",
       "4  MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json       NaN   \n",
       "\n",
       "  zygosity  \n",
       "0      NaN  \n",
       "1      NaN  \n",
       "2      NaN  \n",
       "3      NaN  \n",
       "4      NaN  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proband_data[\"disease\"] = proband_data['disease'].map(lambda x: x.strip('obo:'))\n",
    "proband_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "2e742749-28e6-4f01-96e2-af7d54604295",
   "metadata": {},
   "outputs": [],
   "source": [
    "proband_data_w_disease_labels = pd.merge(proband_data,mondo_lookup, left_on='disease', right_on='Class ID', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "f71802d3-4b39-4f0e-8f62-bbec78a07be3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>type</th>\n",
       "      <th>dc:source</th>\n",
       "      <th>rdfs:label</th>\n",
       "      <th>ageType</th>\n",
       "      <th>ageUnit</th>\n",
       "      <th>ageValue</th>\n",
       "      <th>allele</th>\n",
       "      <th>detectionMethod</th>\n",
       "      <th>firstTestingMethod</th>\n",
       "      <th>...</th>\n",
       "      <th>secondTestingMethod</th>\n",
       "      <th>sex</th>\n",
       "      <th>variant</th>\n",
       "      <th>gene</th>\n",
       "      <th>disease</th>\n",
       "      <th>source_file</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>zygosity</th>\n",
       "      <th>Class ID</th>\n",
       "      <th>Preferred Label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/0708860...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/11709191</td>\n",
       "      <td>Yoshida_Patient MK</td>\n",
       "      <td>AgeAtReport</td>\n",
       "      <td>Years</td>\n",
       "      <td>8.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>The entire coding region (ex 1-22) and intron-...</td>\n",
       "      <td>PCR</td>\n",
       "      <td>...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Male</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>myopathy caused by variation in POMGNT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/352d557...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/26908613</td>\n",
       "      <td>Xu_Proband 2</td>\n",
       "      <td>AgeAtReport</td>\n",
       "      <td>Years</td>\n",
       "      <td>32.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>Reanalysis of WES data revealed the POMGNT1 va...</td>\n",
       "      <td>Genotyping</td>\n",
       "      <td>...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Female</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>myopathy caused by variation in POMGNT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/5f3835c...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/11709191</td>\n",
       "      <td>Yoshida_Patient SA</td>\n",
       "      <td>AgeAtReport</td>\n",
       "      <td>Years</td>\n",
       "      <td>7.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>The entire coding region (ex 1-22) and intron-...</td>\n",
       "      <td>PCR</td>\n",
       "      <td>...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Female</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>myopathy caused by variation in POMGNT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/612230d...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/11709191</td>\n",
       "      <td>Yoshida_Patient YA</td>\n",
       "      <td>AgeAtReport</td>\n",
       "      <td>Years</td>\n",
       "      <td>6.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>The entire coding region (ex 1-22) and intron-...</td>\n",
       "      <td>PCR</td>\n",
       "      <td>...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Male</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>myopathy caused by variation in POMGNT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>https://genegraph.clinicalgenome.org/r/69b2ace...</td>\n",
       "      <td>Proband</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/18195152</td>\n",
       "      <td>Clement_Proband</td>\n",
       "      <td>AgeAtOnset</td>\n",
       "      <td>Years</td>\n",
       "      <td>12.0</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>The POMGNT1 gene was screened for mutations.</td>\n",
       "      <td>Other</td>\n",
       "      <td>...</td>\n",
       "      <td>Sanger sequencing</td>\n",
       "      <td>Female</td>\n",
       "      <td>{'id': 'https://genegraph.clinicalgenome.org/r...</td>\n",
       "      <td>hgnc:19139</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO_0700068</td>\n",
       "      <td>myopathy caused by variation in POMGNT1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  id     type  \\\n",
       "0  https://genegraph.clinicalgenome.org/r/0708860...  Proband   \n",
       "1  https://genegraph.clinicalgenome.org/r/352d557...  Proband   \n",
       "2  https://genegraph.clinicalgenome.org/r/5f3835c...  Proband   \n",
       "3  https://genegraph.clinicalgenome.org/r/612230d...  Proband   \n",
       "4  https://genegraph.clinicalgenome.org/r/69b2ace...  Proband   \n",
       "\n",
       "                                  dc:source          rdfs:label      ageType  \\\n",
       "0  https://pubmed.ncbi.nlm.nih.gov/11709191  Yoshida_Patient MK  AgeAtReport   \n",
       "1  https://pubmed.ncbi.nlm.nih.gov/26908613        Xu_Proband 2  AgeAtReport   \n",
       "2  https://pubmed.ncbi.nlm.nih.gov/11709191  Yoshida_Patient SA  AgeAtReport   \n",
       "3  https://pubmed.ncbi.nlm.nih.gov/11709191  Yoshida_Patient YA  AgeAtReport   \n",
       "4  https://pubmed.ncbi.nlm.nih.gov/18195152     Clement_Proband   AgeAtOnset   \n",
       "\n",
       "  ageUnit  ageValue                                             allele  \\\n",
       "0   Years       8.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "1   Years      32.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "2   Years       7.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "3   Years       6.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "4   Years      12.0  {'id': 'https://genegraph.clinicalgenome.org/r...   \n",
       "\n",
       "                                     detectionMethod firstTestingMethod  ...  \\\n",
       "0  The entire coding region (ex 1-22) and intron-...                PCR  ...   \n",
       "1  Reanalysis of WES data revealed the POMGNT1 va...         Genotyping  ...   \n",
       "2  The entire coding region (ex 1-22) and intron-...                PCR  ...   \n",
       "3  The entire coding region (ex 1-22) and intron-...                PCR  ...   \n",
       "4       The POMGNT1 gene was screened for mutations.              Other  ...   \n",
       "\n",
       "  secondTestingMethod     sex  \\\n",
       "0   Sanger sequencing    Male   \n",
       "1   Sanger sequencing  Female   \n",
       "2   Sanger sequencing  Female   \n",
       "3   Sanger sequencing    Male   \n",
       "4   Sanger sequencing  Female   \n",
       "\n",
       "                                             variant        gene  \\\n",
       "0  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "1  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "2  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "3  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "4  {'id': 'https://genegraph.clinicalgenome.org/r...  hgnc:19139   \n",
       "\n",
       "         disease                                       source_file ethnicity  \\\n",
       "0  MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json       NaN   \n",
       "1  MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json       NaN   \n",
       "2  MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json       NaN   \n",
       "3  MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json       NaN   \n",
       "4  MONDO_0700068  gg_00140591-caa8-4d47-b4ca-3f0577b16d73v2.1.json       NaN   \n",
       "\n",
       "  zygosity       Class ID                          Preferred Label  \n",
       "0      NaN  MONDO_0700068  myopathy caused by variation in POMGNT1  \n",
       "1      NaN  MONDO_0700068  myopathy caused by variation in POMGNT1  \n",
       "2      NaN  MONDO_0700068  myopathy caused by variation in POMGNT1  \n",
       "3      NaN  MONDO_0700068  myopathy caused by variation in POMGNT1  \n",
       "4      NaN  MONDO_0700068  myopathy caused by variation in POMGNT1  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proband_data_w_disease_labels.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "bd9d9504-fd0d-4570-95e5-ba455cf49fe0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 20490 entries, 0 to 20489\n",
      "Data columns (total 24 columns):\n",
      " #   Column                      Non-Null Count  Dtype  \n",
      "---  ------                      --------------  -----  \n",
      " 0   id                          20490 non-null  object \n",
      " 1   type                        20490 non-null  object \n",
      " 2   dc:source                   20490 non-null  object \n",
      " 3   rdfs:label                  20490 non-null  object \n",
      " 4   ageType                     15022 non-null  object \n",
      " 5   ageUnit                     15020 non-null  object \n",
      " 6   ageValue                    12808 non-null  float64\n",
      " 7   allele                      11769 non-null  object \n",
      " 8   detectionMethod             13773 non-null  object \n",
      " 9   firstTestingMethod          16570 non-null  object \n",
      " 10  phenotypeFreeText           11228 non-null  object \n",
      " 11  phenotypes                  14122 non-null  object \n",
      " 12  previousTesting             11858 non-null  object \n",
      " 13  previousTestingDescription  9660 non-null   object \n",
      " 14  secondTestingMethod         8115 non-null   object \n",
      " 15  sex                         20480 non-null  object \n",
      " 16  variant                     20490 non-null  object \n",
      " 17  gene                        20490 non-null  object \n",
      " 18  disease                     20490 non-null  object \n",
      " 19  source_file                 20490 non-null  object \n",
      " 20  ethnicity                   3100 non-null   object \n",
      " 21  zygosity                    973 non-null    object \n",
      " 22  Class ID                    20477 non-null  object \n",
      " 23  Preferred Label             20477 non-null  object \n",
      "dtypes: float64(1), object(23)\n",
      "memory usage: 3.8+ MB\n"
     ]
    }
   ],
   "source": [
    "# Basic stats\n",
    "proband_data_w_disease_labels.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "23c5f936-d984-440e-9942-e7c7eab5ef7a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Index: 12808 entries, 0 to 20489\n",
      "Data columns (total 24 columns):\n",
      " #   Column                      Non-Null Count  Dtype  \n",
      "---  ------                      --------------  -----  \n",
      " 0   id                          12808 non-null  object \n",
      " 1   type                        12808 non-null  object \n",
      " 2   dc:source                   12808 non-null  object \n",
      " 3   rdfs:label                  12808 non-null  object \n",
      " 4   ageType                     12743 non-null  object \n",
      " 5   ageUnit                     12761 non-null  object \n",
      " 6   ageValue                    12808 non-null  float64\n",
      " 7   allele                      7542 non-null   object \n",
      " 8   detectionMethod             9085 non-null   object \n",
      " 9   firstTestingMethod          10880 non-null  object \n",
      " 10  phenotypeFreeText           7745 non-null   object \n",
      " 11  phenotypes                  9888 non-null   object \n",
      " 12  previousTesting             7830 non-null   object \n",
      " 13  previousTestingDescription  6367 non-null   object \n",
      " 14  secondTestingMethod         5374 non-null   object \n",
      " 15  sex                         12808 non-null  object \n",
      " 16  variant                     12808 non-null  object \n",
      " 17  gene                        12808 non-null  object \n",
      " 18  disease                     12808 non-null  object \n",
      " 19  source_file                 12808 non-null  object \n",
      " 20  ethnicity                   2267 non-null   object \n",
      " 21  zygosity                    678 non-null    object \n",
      " 22  Class ID                    12808 non-null  object \n",
      " 23  Preferred Label             12808 non-null  object \n",
      "dtypes: float64(1), object(23)\n",
      "memory usage: 2.4+ MB\n"
     ]
    }
   ],
   "source": [
    "# Check for temporal data\n",
    "temporal_data = proband_data_w_disease_labels[proband_data_w_disease_labels[\"ageValue\"].notnull()]\n",
    "temporal_data.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "74dd941b-c454-437c-adde-266d2660c6f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.DataFrame(temporal_data['Preferred Label'].value_counts()).reset_index().to_csv(\"./output/disease_w_temoral_info.csv\", index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "30091692-d76e-4455-8937-1cb4e53de594",
   "metadata": {},
   "outputs": [],
   "source": [
    "proband_data_w_disease_labels.to_csv(\"./output/all_proband_data.csv\",index= False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
