{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e23e7eff-1873-4da4-baeb-436bf168f904",
   "metadata": {},
   "outputs": [],
   "source": [
    "import collections\n",
    "import itertools\n",
    "import pyld\n",
    "import json\n",
    "import phenopackets as pps\n",
    "import pyhpo\n",
    "import jsonpath_ng\n",
    "from google.protobuf.json_format import MessageToJson\n",
    "from google.protobuf.timestamp_pb2 import Timestamp\n",
    "import glob\n",
    "import gzip\n",
    "import csv\n",
    "import statistics as stat\n",
    "import logging"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "dc9322cd-b320-4465-9d23-6918495be772",
   "metadata": {},
   "outputs": [],
   "source": [
    "logger = logging.getLogger(__name__)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1a3a277",
   "metadata": {},
   "source": [
    "JSON-LD, being very RDF-y, will represent 0..* fields as single items if there's just one, and lists if more than one. Sometimes it's more convenient to be consistent."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "02752d2d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def ensure_list(value):\n",
    "    if isinstance(value, list):\n",
    "        return value\n",
    "    return [value]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ce7ed33a-cd56-48ed-81f2-5db3921d67f1",
   "metadata": {},
   "source": [
    "A dictionary of MONDO ids to labels (should probably use ClinGen-preferred when available)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6f2976cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "with gzip.open('MONDO.csv.gz', 'rt') as csvf:\n",
    "    '''Keys will be like \"MONDO_0044647\", values like \"kyphosis-lateral tongue atrophy-myofibrillar myopathy syndrome\"'''\n",
    "    reader = csv.DictReader(csvf)\n",
    "    class_prefix = 'http://purl.obolibrary.org/obo/'\n",
    "    mondo_lookup = dict((row['Class ID'][len(class_prefix):], row['Preferred Label']) for row in reader if row['Class ID'].startswith(class_prefix))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4213bf84-3ff0-4a3b-adc2-45e2bf0d602e",
   "metadata": {},
   "source": [
    "load the HPO ontology using pyhpo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d0ef5530",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2025-01-16\n",
      "The Ontology has been built before already\n"
     ]
    }
   ],
   "source": [
    "pyhpo.Ontology()\n",
    "print(pyhpo.Ontology.version())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "97634082-4602-40d1-8ee2-f9b198eab866",
   "metadata": {},
   "source": [
    "We will need this later for protobuf representation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "ec518ccb-5548-4316-af87-2bdb46fd13b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "timestamp = Timestamp()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a477f3fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "j = json.load(open('input/gene-validity-202512/gg_02212618-0725-43db-9a4c-7929faeb878av1.0.json'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "19ca09a0-3e03-4432-a3f9-511bed16f2db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'obo:MONDO_0013135'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "j['subject']['disease']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b1a259ca",
   "metadata": {},
   "source": [
    "The idea behind using JSON-LD framing is two-fold:\n",
    "* It gives us a quick way to get down to the probands\n",
    "* By specifying `embed` of `@always`, we should be able to avoid having to keep our own dictionary of objects (*for the most part*... pyld does not appear to handle re-embeddeding `dc:source` when needed below)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "33c71ba1-0a59-4564-925c-b8c99c2d90ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "frame = {'@context': j['@context'], '@type': 'Proband', '@embed': '@always'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f02fa512-0293-42bf-bad3-103249cb63d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "framed = pyld.jsonld.frame(j, frame)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "65df1d60-acb2-417e-9c9c-3684558ee480",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(framed['@graph'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "0f9a247e-2f98-48e5-a856-30788b3081dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['id', 'type', 'dc:source', 'rdfs:label', 'ageType', 'ageUnit', 'ageValue', 'allele', 'detectionMethod', 'firstTestingMethod', 'phenotypeFreeText', 'previousTestingDescription', 'secondTestingMethod', 'sex', 'variant'])"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "framed['@graph'][0].keys()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bc8ad1b8",
   "metadata": {},
   "source": [
    "*FIXME*: not really sure what to use for `resources`, so just going with what is in the phenopacket store for now..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "a436ef1a",
   "metadata": {
    "jupyter": {
     "source_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "resources = [\n",
    "  pps.Resource(**resource) for resource in [\n",
    "    {\n",
    "      \"id\": \"geno\",\n",
    "      \"name\": \"Genotype Ontology\",\n",
    "      \"url\": \"http://purl.obolibrary.org/obo/geno.owl\",\n",
    "      \"version\": \"2022-03-05\",\n",
    "      \"namespace_prefix\": \"GENO\",\n",
    "      \"iri_prefix\": \"http://purl.obolibrary.org/obo/GENO_\"\n",
    "    },\n",
    "    {\n",
    "      \"id\": \"hgnc\",\n",
    "      \"name\": \"HUGO Gene Nomenclature Committee\",\n",
    "      \"url\": \"https://www.genenames.org\",\n",
    "      \"version\": \"06/01/23\",\n",
    "      \"namespace_prefix\": \"HGNC\",\n",
    "      \"iri_prefix\": \"https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/\"\n",
    "    },\n",
    "    {\n",
    "      \"id\": \"omim\",\n",
    "      \"name\": \"An Online Catalog of Human Genes and Genetic Disorders\",\n",
    "      \"url\": \"https://www.omim.org\",\n",
    "      \"version\": \"January 4, 2023\",\n",
    "      \"namespace_prefix\": \"OMIM\",\n",
    "      \"iri_prefix\": \"https://www.omim.org/entry/\"\n",
    "    },\n",
    "    {\n",
    "      \"id\": \"so\",\n",
    "      \"name\": \"Sequence types and features ontology\",\n",
    "      \"url\": \"http://purl.obolibrary.org/obo/so.obo\",\n",
    "      \"version\": \"2021-11-22\",\n",
    "      \"namespace_prefix\": \"SO\",\n",
    "      \"iri_prefix\": \"http://purl.obolibrary.org/obo/SO_\"\n",
    "    },\n",
    "    {\n",
    "      \"id\": \"hp\",\n",
    "      \"name\": \"human phenotype ontology\",\n",
    "      \"url\": \"http://purl.obolibrary.org/obo/hp.owl\",\n",
    "      \"version\": \"2024-07-01\",\n",
    "      \"namespace_prefix\": \"HP\",\n",
    "      \"iri_prefix\": \"http://purl.obolibrary.org/obo/HP_\"\n",
    "    }\n",
    "  ]\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "f22edf68-ae12-40eb-ad19-b7c33b4c8328",
   "metadata": {},
   "outputs": [],
   "source": [
    "def phenotype_element_from_id(hpoid, **kwargs):\n",
    "    try:\n",
    "        term = pyhpo.Ontology.get_hpo_object(hpoid)\n",
    "    except:\n",
    "        logger.error(f'unable to find term for {hpoid}')\n",
    "    return pps.PhenotypicFeature(type=pps.OntologyClass(id=term.id, label=term.name), **kwargs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "252a452f-d9b7-44ee-9ca2-513c0fd6863a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'id': 'https://genegraph.clinicalgenome.org/r/5d76c7de-d884-409c-858d-a9c7995aa5cf',\n",
       " 'pmids': '',\n",
       " 'label': '1875',\n",
       " 'phenotypes': [{'id': 'HP:0004313',\n",
       "   'label': 'Decreased circulating antibody concentration'}]}"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def obo_to_labeled_phenotype(obo):\n",
    "    try:\n",
    "        term = pyhpo.Ontology.get_hpo_object(obo.replace('obo:HP_', 'HP:'))\n",
    "        return {'id': term.id, 'label': term.name}\n",
    "    except:\n",
    "        return {'id': obo}\n",
    "\n",
    "dcsource_path = jsonpath_ng.parse('$..\"dc:source\"')\n",
    "\n",
    "def id_or_value(x):\n",
    "    if isinstance(x, dict):\n",
    "        return x.get('id', '')\n",
    "    return x\n",
    "\n",
    "def find_sources(proband):\n",
    "    try:\n",
    "        dc_sources = [proband['dc:source']]\n",
    "    except KeyError:\n",
    "        try:\n",
    "          found = reversed(dcsource_path.find(proband))\n",
    "          # Note: there is some hacky handling of non-embeded dc:source by testing\n",
    "          #   for a string and ignoring (since it *should* be embedded elsewhere, then\n",
    "          #   but this will probably break when there are multiple probands from some\n",
    "          #   source in same file. Maybe pyld's @embed: @always will work better\n",
    "          #   some day...\n",
    "          sources_by_id = dict((x.value.get('id', 'UNKNOWN'), x.value)\n",
    "                               for x in found\n",
    "                               if not isinstance(x.value, str))\n",
    "          dc_sources = list(sources_by_id.values())\n",
    "        except Exception as e:\n",
    "            print(f'ERROR finding dc:source with {proband=}')\n",
    "            print(e)\n",
    "            dc_sources = []\n",
    "    return dc_sources\n",
    "\n",
    "def proband_phenotype_report(proband):\n",
    "    phenotypes = [obo_to_labeled_phenotype(p) for p in ensure_list(proband.get('phenotypes', []))]\n",
    "    sources = find_sources(proband)\n",
    "    #FIXME\n",
    "    try:\n",
    "        pmids = ','.join(set(s['id'].replace('https://pubmed.ncbi.nlm.nih.gov/', '') for s in sources))\n",
    "    except:\n",
    "        pmids = ''\n",
    "    return {\n",
    "        'id': proband['id'],\n",
    "        'pmids': pmids,\n",
    "        'label': proband['rdfs:label'],\n",
    "        'phenotypes': phenotypes,\n",
    "    }\n",
    "\n",
    "proband_phenotype_report(framed['@graph'][4])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "63754a99-120c-4201-88ab-37793e17215c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'id': 'https://genegraph.clinicalgenome.org/r/5d76c7de-d884-409c-858d-a9c7995aa5cf',\n",
       " 'type': 'Proband',\n",
       " 'dc:source': 'https://pubmed.ncbi.nlm.nih.gov/19804848',\n",
       " 'rdfs:label': '1875',\n",
       " 'ageType': 'AgeAtOnset',\n",
       " 'ageUnit': 'Months',\n",
       " 'ageValue': 4,\n",
       " 'allele': {'id': 'https://genegraph.clinicalgenome.org/r/ce6508ac-b270-4f95-a36a-a356ecdb286f',\n",
       "  'type': 'https://terms.ga4gh.org/VariationDescriptor',\n",
       "  'http://www.w3.org/2004/02/skos/core#prefLabel': 'NM_006949.4(STXBP2):c.1430C>T (p.Pro477Leu)',\n",
       "  'https://terms.ga4gh.org/CanonicalReference': {'id': 'http://reg.genome.network/allele/CA254261'}},\n",
       " 'detectionMethod': 'Whole-genome homozygosity mapping. STXBP2 was then PCR amplified and there was Sanger sequencing of coding exons and adjacent intronic sequences.',\n",
       " 'firstTestingMethod': 'Homozygosity mapping',\n",
       " 'phenotypeFreeText': '5/8 HLH criteria met',\n",
       " 'phenotypes': 'obo:HP_0004313',\n",
       " 'previousTestingDescription': 'Screened for mutations in PRF1, UNC13D, and STX11.',\n",
       " 'secondTestingMethod': 'Sanger sequencing',\n",
       " 'sex': 'Female',\n",
       " 'variant': {'id': 'https://genegraph.clinicalgenome.org/r/3ad9ddfd-399f-4820-905c-31f24c0a72be_variant_evidence_item',\n",
       "  'type': 'VariantObservation',\n",
       "  'dc:source': 'https://pubmed.ncbi.nlm.nih.gov/19804848',\n",
       "  'allele': {'id': 'https://genegraph.clinicalgenome.org/r/ce6508ac-b270-4f95-a36a-a356ecdb286f',\n",
       "   'type': 'https://terms.ga4gh.org/VariationDescriptor',\n",
       "   'http://www.w3.org/2004/02/skos/core#prefLabel': 'NM_006949.4(STXBP2):c.1430C>T (p.Pro477Leu)',\n",
       "   'https://terms.ga4gh.org/CanonicalReference': {'id': 'http://reg.genome.network/allele/CA254261'}},\n",
       "  'alleleOrigin': {'id': 'cg:AlleleOrigin'},\n",
       "  'zygosity': {'id': 'cg:Homozygous'}}}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "framed['@graph'][4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "bba97d72-345b-4f96-9d36-d5a71b839eeb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['https://pubmed.ncbi.nlm.nih.gov/19804848']"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "find_sources(framed['@graph'][0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "644d2f5c-6bd8-4fe9-80ef-d95e20cdca34",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "  \"id\": \"https://genegraph.clinicalgenome.org/r/5d76c7de-d884-409c-858d-a9c7995aa5cf\",\n",
      "  \"subject\": {\n",
      "    \"id\": \"19804848:1875\"\n",
      "  },\n",
      "  \"phenotypicFeatures\": [\n",
      "    {\n",
      "      \"type\": {\n",
      "        \"id\": \"HP:0004313\",\n",
      "        \"label\": \"Decreased circulating antibody concentration\"\n",
      "      },\n",
      "      \"evidence\": [\n",
      "        {\n",
      "          \"evidenceCode\": {\n",
      "            \"id\": \"ECO:0006017\",\n",
      "            \"label\": \"author statement from published clinical study used in manual assertion\"\n",
      "          },\n",
      "          \"reference\": {\n",
      "            \"id\": \"PMID:19804848\",\n",
      "            \"reference\": \"https://pubmed.ncbi.nlm.nih.gov/19804848\"\n",
      "          }\n",
      "        }\n",
      "      ]\n",
      "    }\n",
      "  ],\n",
      "  \"metaData\": {\n",
      "    \"createdBy\": \"Automated import from ClinGen GCI data\",\n",
      "    \"resources\": [\n",
      "      {\n",
      "        \"id\": \"geno\",\n",
      "        \"name\": \"Genotype Ontology\",\n",
      "        \"url\": \"http://purl.obolibrary.org/obo/geno.owl\",\n",
      "        \"version\": \"2022-03-05\",\n",
      "        \"namespacePrefix\": \"GENO\",\n",
      "        \"iriPrefix\": \"http://purl.obolibrary.org/obo/GENO_\"\n",
      "      },\n",
      "      {\n",
      "        \"id\": \"hgnc\",\n",
      "        \"name\": \"HUGO Gene Nomenclature Committee\",\n",
      "        \"url\": \"https://www.genenames.org\",\n",
      "        \"version\": \"06/01/23\",\n",
      "        \"namespacePrefix\": \"HGNC\",\n",
      "        \"iriPrefix\": \"https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/\"\n",
      "      },\n",
      "      {\n",
      "        \"id\": \"omim\",\n",
      "        \"name\": \"An Online Catalog of Human Genes and Genetic Disorders\",\n",
      "        \"url\": \"https://www.omim.org\",\n",
      "        \"version\": \"January 4, 2023\",\n",
      "        \"namespacePrefix\": \"OMIM\",\n",
      "        \"iriPrefix\": \"https://www.omim.org/entry/\"\n",
      "      },\n",
      "      {\n",
      "        \"id\": \"so\",\n",
      "        \"name\": \"Sequence types and features ontology\",\n",
      "        \"url\": \"http://purl.obolibrary.org/obo/so.obo\",\n",
      "        \"version\": \"2021-11-22\",\n",
      "        \"namespacePrefix\": \"SO\",\n",
      "        \"iriPrefix\": \"http://purl.obolibrary.org/obo/SO_\"\n",
      "      },\n",
      "      {\n",
      "        \"id\": \"hp\",\n",
      "        \"name\": \"human phenotype ontology\",\n",
      "        \"url\": \"http://purl.obolibrary.org/obo/hp.owl\",\n",
      "        \"version\": \"2024-07-01\",\n",
      "        \"namespacePrefix\": \"HP\",\n",
      "        \"iriPrefix\": \"http://purl.obolibrary.org/obo/HP_\"\n",
      "      }\n",
      "    ],\n",
      "    \"phenopacketSchemaVersion\": \"2.0\",\n",
      "    \"externalReferences\": [\n",
      "      {\n",
      "        \"id\": \"PMID:19804848\",\n",
      "        \"reference\": \"https://pubmed.ncbi.nlm.nih.gov/19804848\"\n",
      "      }\n",
      "    ]\n",
      "  }\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "def proband_to_phenopacket(proband):\n",
    "    references = []\n",
    "    for dc_source in find_sources(proband):\n",
    "        # until/unless genegraph provides dc:source as an object again instead of just an id/reference...\n",
    "        #pmid = dc_source['id'].replace('https://pubmed.ncbi.nlm.nih.gov/', '')\n",
    "        #reference = pps.ExternalReference(id=f'PMID:{pmid}', reference=dc_source['id'], description=dc_source['dc:title'])\n",
    "        pmid = dc_source.replace('https://pubmed.ncbi.nlm.nih.gov/', '')\n",
    "        reference = pps.ExternalReference(id=f'PMID:{pmid}', reference=dc_source)\n",
    "        referenceEvidence = pps.Evidence(reference=reference, evidence_code=pps.OntologyClass(id=\"ECO:0006017\", label=\"author statement from published clinical study used in manual assertion\"))\n",
    "        references.append(reference)\n",
    "    phenotypes = [phenotype_element_from_id(p.replace('obo:HP_', 'HP:'), evidence=[referenceEvidence]) for p in ensure_list(proband.get('phenotypes', []))]\n",
    "    metadata = pps.MetaData(external_references=references,\n",
    "                            created=timestamp.GetCurrentTime(),\n",
    "                            created_by=\"Automated import from ClinGen GCI data\",\n",
    "                            phenopacket_schema_version=\"2.0\",\n",
    "                            resources=resources)\n",
    "    subject_args = {'id': f'{pmid}:{proband['rdfs:label']}'}\n",
    "    try:\n",
    "        subject_args['sex'] = pps.Sex.Value(proband['sex'])\n",
    "    except:\n",
    "        subject_args['sex'] = pps.Sex.UNKNOWN_SEX\n",
    "    individual = pps.Individual(**subject_args)\n",
    "    phenopacket = pps.Phenopacket(id=proband['id'], subject=individual, phenotypic_features=phenotypes, meta_data=metadata)\n",
    "    return phenopacket\n",
    "\n",
    "print(MessageToJson(proband_to_phenopacket(framed['@graph'][4])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "e6d461d4-6335-44a8-8e1c-d2d4c48897f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3387"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "inputs = sorted(glob.glob('input/gene-validity-202512/*.json'))[::-1]\n",
    "len(inputs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc2e40ec-58f9-45ff-9e31-f7524fd34193",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "proband_summaries = {}\n",
    "for f in inputs:\n",
    "    with open(f) as jf:\n",
    "        j = json.load(jf)\n",
    "    framed = pyld.jsonld.frame(j, frame)\n",
    "    for proband in framed.get('@graph', []):\n",
    "        if proband['id'] in proband_summaries:\n",
    "            print(f'WARNING: more than one record for {proband['id']}, not sure which one to use!')\n",
    "        summary = proband_phenotype_report(proband)\n",
    "        summary['evidenceStrength'] = j.get('evidenceStrength', '')\n",
    "        disease_ids = [x.replace('obo:', '').replace(':', '_') for x in ensure_list(j.get('subject', {}).get('disease', 'unknown'))]\n",
    "        summary['disease'] = [{'id': did, 'label': mondo_lookup.get(did, '')} for did in disease_ids]\n",
    "        proband_summaries[proband['id']] = summary\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "29396cad-ad7c-442c-bb92-702e92a4af86",
   "metadata": {},
   "outputs": [],
   "source": [
    "probands_with_phenotypes = [ps for ps in proband_summaries.values() if len(ps['phenotypes'])> 0]\n",
    "print(f'Probands with phenotypes: {len(probands_with_phenotypes)}')\n",
    "print(f'Probands with phenotypes where evidence at least moderate: {len([x for x in probands_with_phenotypes if x.get('evidenceStrength') in ('Moderate', 'Strong', 'Definitive')])}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "165000cb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "collections.Counter([x.get('disease', [{}])[0].get('label', '') for x in probands_with_phenotypes])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6fe8632",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('probands_with_phenotypes.tsv', 'wt') as ouf:\n",
    "    print('\\t'.join(('evidence_strength', 'evidence_id', 'pmid', 'label', 'phenotypes', 'phenotype_labels', 'diseases', 'disease_labels')), file=ouf)\n",
    "    for line in probands_with_phenotypes:\n",
    "        phenotypes = '|'.join(x['id'] for x in line.get('phenotypes'))\n",
    "        phenotype_labels = '|'.join(x.get('label', '') for x in line.get('phenotypes'))\n",
    "        diseases = '|'.join(x['id'] for x in line.get('disease'))\n",
    "        disease_labels = '|'.join(x['label'] for x in line.get('disease'))\n",
    "        print('\\t'.join((\n",
    "            line['evidenceStrength'],\n",
    "            line['id'],\n",
    "            line['pmids'],\n",
    "            line['label'],\n",
    "            phenotypes,\n",
    "            phenotype_labels,\n",
    "            diseases,\n",
    "            disease_labels,\n",
    "        )), file=ouf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e7fc3fde-29a4-4368-b992-06185ac23ed0",
   "metadata": {},
   "outputs": [],
   "source": [
    "phenotype_counts = [len(p['phenotypes']) for p in proband_summaries.values()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "16323ea2-fdc4-434e-9a45-c5bde0d14ab6",
   "metadata": {},
   "outputs": [],
   "source": [
    "countdata = phenotype_counts\n",
    "print(f'Among all {len(countdata)} probands:')\n",
    "print(f'mean: {stat.mean(countdata)}')\n",
    "print(f'stdev: {stat.stdev(countdata)}')\n",
    "print(f'median: {stat.median(countdata)}')\n",
    "print(f'max: {max(countdata)}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b521c22a-b58c-4755-9135-f92948ae9593",
   "metadata": {},
   "outputs": [],
   "source": [
    "countdata = [pc for pc in phenotype_counts if pc != 0]\n",
    "print(f'Among {len(countdata)} probands with at least one phenotype reported:')\n",
    "print(f'mean: {stat.mean(countdata)}')\n",
    "print(f'stdev: {stat.stdev(countdata)}')\n",
    "print(f'median: {stat.median(countdata)}')\n",
    "print(f'max: {max(countdata)}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e6d7017e-f29e-40c8-8530-7255f5ccd5f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "terms_by_count = collections.Counter()\n",
    "for p in proband_summaries.values():\n",
    "    terms_by_count.update(pheno.get('label', pheno.get('id', 'UNKNOWN')) for pheno in p['phenotypes'])\n",
    "terms_by_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b2d48e3-13cd-4eb5-bc94-82be7f5fdf09",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
